{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genomic Data Processing Pipeline\n",
    "This notebook processes genomic VCF data using Hail for machine learning preparation.\n",
    "## Setup and Requirements\n",
    "- Hail\n",
    "- Pandas\n",
    "- Python 3.9+\n",
    "## Pipeline Overview\n",
    "1. Data Loading\n",
    "2. Quality Control\n",
    "3. Feature Extraction\n",
    "## 1. Initial Setup\n",
    "First, we'll import our required libraries and set up logging utilities."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create a temporary directory path"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Initialize Hail for local processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "    <style>\n",
       "        .bk-notebook-logo {\n",
       "            display: block;\n",
       "            width: 20px;\n",
       "            height: 20px;\n",
       "            background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAABx0RVh0U29mdHdhcmUAQWRvYmUgRmlyZXdvcmtzIENTNui8sowAAAOkSURBVDiNjZRtaJVlGMd/1/08zzln5zjP1LWcU9N0NkN8m2CYjpgQYQXqSs0I84OLIC0hkEKoPtiH3gmKoiJDU7QpLgoLjLIQCpEsNJ1vqUOdO7ppbuec5+V+rj4ctwzd8IIbbi6u+8f1539dt3A78eXC7QizUF7gyV1fD1Yqg4JWz84yffhm0qkFqBogB9rM8tZdtwVsPUhWhGcFJngGeWrPzHm5oaMmkfEg1usvLFyc8jLRqDOMru7AyC8saQr7GG7f5fvDeH7Ej8CM66nIF+8yngt6HWaKh7k49Soy9nXurCi1o3qUbS3zWfrYeQDTB/Qj6kX6Ybhw4B+bOYoLKCC9H3Nu/leUTZ1JdRWkkn2ldcCamzrcf47KKXdAJllSlxAOkRgyHsGC/zRday5Qld9DyoM4/q/rUoy/CXh3jzOu3bHUVZeU+DEn8FInkPBFlu3+nW3Nw0mk6vCDiWg8CeJaxEwuHS3+z5RgY+YBR6V1Z1nxSOfoaPa4LASWxxdNp+VWTk7+4vzaou8v8PN+xo+KY2xsw6une2frhw05CTYOmQvsEhjhWjn0bmXPjpE1+kplmmkP3suftwTubK9Vq22qKmrBhpY4jvd5afdRA3wGjFAgcnTK2s4hY0/GPNIb0nErGMCRxWOOX64Z8RAC4oCXdklmEvcL8o0BfkNK4lUg9HTl+oPlQxdNo3Mg4Nv175e/1LDGzZen30MEjRUtmXSfiTVu1kK8W4txyV6BMKlbgk3lMwYCiusNy9fVfvvwMxv8Ynl6vxoByANLTWplvuj/nF9m2+PDtt1eiHPBr1oIfhCChQMBw6Aw0UulqTKZdfVvfG7VcfIqLG9bcldL/+pdWTLxLUy8Qq38heUIjh4XlzZxzQm19lLFlr8vdQ97rjZVOLf8nclzckbcD4wxXMidpX30sFd37Fv/GtwwhzhxGVAprjbg0gCAEeIgwCZyTV2Z1REEW8O4py0wsjeloKoMr6iCY6dP92H6Vw/oTyICIthibxjm/DfN9lVz8IqtqKYLUXfoKVMVQVVJOElGjrnnUt9T9wbgp8AyYKaGlqingHZU/uG2NTZSVqwHQTWkx9hxjkpWDaCg6Ckj5qebgBVbT3V3NNXMSiWSDdGV3hrtzla7J+duwPOToIg42ChPQOQjspnSlp1V+Gjdged7+8UN5CRAV7a5EdFNwCjEaBR27b3W890TE7g24NAP/mMDXRWrGoFPQI9ls/MWO2dWFAar/xcOIImbbpA3zgAAAABJRU5ErkJggg==);\n",
       "        }\n",
       "    </style>\n",
       "    <div>\n",
       "        <a href=\"https://bokeh.org\" target=\"_blank\" class=\"bk-notebook-logo\"></a>\n",
       "        <span id=\"f4e6b08c-f83f-45bc-a5c6-5eaee23cfd87\">Loading BokehJS ...</span>\n",
       "    </div>\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": "(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\nconst JS_MIME_TYPE = 'application/javascript';\n  const HTML_MIME_TYPE = 'text/html';\n  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n  const CLASS_NAME = 'output_bokeh rendered_html';\n\n  /**\n   * Render data to the DOM node\n   */\n  function render(props, node) {\n    const script = document.createElement(\"script\");\n    node.appendChild(script);\n  }\n\n  /**\n   * Handle when an output is cleared or removed\n   */\n  function handleClearOutput(event, handle) {\n    function drop(id) {\n      const view = Bokeh.index.get_by_id(id)\n      if (view != null) {\n        view.model.document.clear()\n        Bokeh.index.delete(view)\n      }\n    }\n\n    const cell = handle.cell;\n\n    const id = cell.output_area._bokeh_element_id;\n    const server_id = cell.output_area._bokeh_server_id;\n\n    // Clean up Bokeh references\n    if (id != null) {\n      drop(id)\n    }\n\n    if (server_id !== undefined) {\n      // Clean up Bokeh references\n      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n      cell.notebook.kernel.execute(cmd_clean, {\n        iopub: {\n          output: function(msg) {\n            const id = msg.content.text.trim()\n            drop(id)\n          }\n        }\n      });\n      // Destroy server and session\n      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n      cell.notebook.kernel.execute(cmd_destroy);\n    }\n  }\n\n  /**\n   * Handle when a new output is added\n   */\n  function handleAddOutput(event, handle) {\n    const output_area = handle.output_area;\n    const output = handle.output;\n\n    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n      return\n    }\n\n    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n\n    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n      // store reference to embed id on output_area\n      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n    }\n    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n      const bk_div = document.createElement(\"div\");\n      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n      const script_attrs = bk_div.children[0].attributes;\n      for (let i = 0; i < script_attrs.length; i++) {\n        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n      }\n      // store reference to server id on output_area\n      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n    }\n  }\n\n  function register_renderer(events, OutputArea) {\n\n    function append_mime(data, metadata, element) {\n      // create a DOM node to render to\n      const toinsert = this.create_output_subarea(\n        metadata,\n        CLASS_NAME,\n        EXEC_MIME_TYPE\n      );\n      this.keyboard_manager.register_events(toinsert);\n      // Render to node\n      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n      render(props, toinsert[toinsert.length - 1]);\n      element.append(toinsert);\n      return toinsert\n    }\n\n    /* Handle when an output is cleared or removed */\n    events.on('clear_output.CodeCell', handleClearOutput);\n    events.on('delete.Cell', handleClearOutput);\n\n    /* Handle when a new output is added */\n    events.on('output_added.OutputArea', handleAddOutput);\n\n    /**\n     * Register the mime type and append_mime function with output_area\n     */\n    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n      /* Is output safe? */\n      safe: true,\n      /* Index of renderer in `output_area.display_order` */\n      index: 0\n    });\n  }\n\n  // register the mime type if in Jupyter Notebook environment and previously unregistered\n  if (root.Jupyter !== undefined) {\n    const events = require('base/js/events');\n    const OutputArea = require('notebook/js/outputarea').OutputArea;\n\n    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n      register_renderer(events, OutputArea);\n    }\n  }\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    const el = document.getElementById(\"f4e6b08c-f83f-45bc-a5c6-5eaee23cfd87\");\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.3.4.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.3.4.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.3.4.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.3.4.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.3.4.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n          for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\nif (force === true) {\n        display_loaded();\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(\"f4e6b08c-f83f-45bc-a5c6-5eaee23cfd87\")).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));",
      "application/vnd.bokehjs_load.v0+json": ""
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Initializing Hail with default parameters...\n",
      "Running on Apache Spark version 3.5.4\n",
      "SparkUI available at http://LT2058-20241028.SHTC.com:4040\n",
      "Welcome to\n",
      "     __  __     <>__\n",
      "    / /_/ /__  __/ /\n",
      "   / __  / _ `/ / /\n",
      "  /_/ /_/\\_,_/_/_/   version 0.2.133-4c60fddb171a\n",
      "LOGGING: writing to c:\\Users\\jmcdonald\\source\\PersonalRepos\\genome-analysis-ml\\notebooks\\hail-20250112-1231-0.2.133-4c60fddb171a.log\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Hail is already initialized\n"
     ]
    }
   ],
   "source": [
    "import hail as hl\n",
    "import os\n",
    "import tempfile\n",
    "import time\n",
    "from urllib.parse import urlparse, urlunparse\n",
    "\n",
    "def log_step(message):\n",
    "    \"\"\"Utility function to log steps with timestamps\"\"\"\n",
    "    print(f\"\\n[{time.strftime('%H:%M:%S')}] {message}\")\n",
    "\n",
    "def _get_local_tmpdir():\n",
    "    tmp_path = tempfile.gettempdir()\n",
    "    if os.name == 'nt':  # Windows\n",
    "        tmp_url = f'file:///{tmp_path.replace(os.sep, \"/\")}'\n",
    "    else:  # Unix-like systems\n",
    "        tmp_url = f'file://{tmp_path}'\n",
    "    r = urlparse(tmp_url)\n",
    "    if not r.scheme:\n",
    "        r = r._replace(scheme='file')\n",
    "    return urlunparse(r)\n",
    "\n",
    "try:\n",
    "    # Check if Hail is already initialized\n",
    "    if hl.eval(hl.literal(1)) == 1:\n",
    "        print(\"Hail is already initialized\")\n",
    "\n",
    "except Exception:\n",
    "    # If Hail is not initialized or there's an error, initialize it\n",
    "    local_tmpdir = _get_local_tmpdir()\n",
    "    print(f\"Initializing Hail with temporary directory: {local_tmpdir}\")\n",
    "    \n",
    "    hl.init(\n",
    "        default_reference=\"GRCh38\",\n",
    "        local_tmpdir=local_tmpdir\n",
    "    )\n",
    "    print(\"Hail initialized successfully\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking if file exists: True\n",
      "\n",
      "[12:32:40] Loading VCF file...\n",
      "Error counting variants: HailException: Invalid locus 'chr13:48379904' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\n",
      "\n",
      "Java stack trace:\n",
      "org.apache.spark.SparkException: Job aborted due to stage failure: Task 12 in stage 1.0 failed 1 times, most recent failure: Lost task 12.0 in stage 1.0 (TID 38) (LT2058-20241028.SHTC.com executor driver): is.hail.utils.HailException: file:/C:/Users/jmcdonald/Desktop/Projects/The Genome Project/gnomad.exomes.v4.1.sites.chr13.vcf.bgz:offset 107083150656189: error while parsing line\n",
      "chr13\t48379904\t.\tG\tA\t.\tAC0\tAC=0;AN=614612;AF=0.00000;AC_XX=0;AF_XX=0.00000;AN_XX=297944;nhomalt_XX=0;AC_XY=0;AF_XY=0.00000;AN_XY=316668;nhomalt_XY=0;nhomalt=0;AC_afr_XX=0;AF_afr_XX=0.00000;AN_afr_XX=8146;nhomalt_afr_XX=0;AC_afr_XY=0;AF_afr_XY=0.00000;AN_afr_XY=6494;nhomalt_afr_XY=0;AC_afr=0;AF_afr=0.00000;AN_afr=14640;nhomalt_afr=0;AC_amr_XX=0;AF_amr_XX=0.00000;AN_amr_XX=7498;nhomalt_amr_XX=0;AC_amr_XY=0;AF_amr_XY=0.00000;AN_amr_XY=6868;nhomalt_amr_XY=0;AC_amr=0;AF_amr=0.00000;AN_amr=14366;nhomalt_amr=0;AC_asj_XX=0;AF_asj_XX=0.00000;AN_asj_XX=6192;nhomalt_asj_XX=0;AC_asj_XY=0;AF_asj_XY=0.00000;AN_asj_XY=6840;nhomalt_asj_XY=0;AC_asj=0;AF_asj=0.00000;AN_asj=13032;nhomalt_asj=0;AC_eas_XX=0;AF_eas_XX=0.00000;AN_eas_XX=11742;nhomalt_eas_XX=0;AC_eas_XY=0;AF_eas_XY=0.00000;AN_eas_XY=11538;nhomalt_eas_XY=0;AC_eas=0;AF_eas=0.00000;AN_eas=23280;nhomalt_eas=0;AC_fin_XX=0;AF_fin_XX=0.00000;AN_fin_XX=11334;nhomalt_fin_XX=0;AC_fin_XY=0;AF_fin_XY=0.00000;AN_fin_XY=10082;nhomalt_fin_XY=0;AC_fin=0;AF_fin=0.00000;AN_fin=21416;nhomalt_fin=0;AC_mid_XX=0;AF_mid_XX=0.00000;AN_mid_XX=814;nhomalt_mid_XX=0;AC_mid_XY=0;AF_mid_XY=0.00000;AN_mid_XY=1260;nhomalt_mid_XY=0;AC_mid=0;AF_mid=0.00000;AN_mid=2074;nhomalt_mid=0;AC_nfe_XX=0;AF_nfe_XX=0.00000;AN_nfe_XX=227204;nhomalt_nfe_XX=0;AC_nfe_XY=0;AF_nfe_XY=0.00000;AN_nfe_XY=225072;nhomalt_nfe_XY=0;AC_nfe=0;AF_nfe=0.00000;AN_nfe=452276;nhomalt_nfe=0;AC_non_ukb_XX=0;AF_non_ukb_XX=0.00000;AN_non_ukb_XX=165874;nhomalt_non_ukb_XX=0;AC_non_ukb_XY=0;AF_non_ukb_XY=0.00000;AN_non_ukb_XY=198072;nhomalt_non_ukb_XY=0;AC_non_ukb=0;AF_non_ukb=0.00000;AN_non_ukb=363946;nhomalt_non_ukb=0;AC_non_ukb_afr_XX=0;AF_non_ukb_afr_XX=0.00000;AN_non_ukb_afr_XX=5548;nhomalt_non_ukb_afr_XX=0;AC_non_ukb_afr_XY=0;AF_non_ukb_afr_XY=0.00000;AN_non_ukb_afr_XY=4448;nhomalt_non_ukb_afr_XY=0;AC_non_ukb_afr=0;AF_non_ukb_afr=0.00000;AN_non_ukb_afr=9996;nhomalt_non_ukb_afr=0;AC_non_ukb_amr_XX=0;AF_non_ukb_amr_XX=0.00000;AN_non_ukb_amr_XX=7330;nhomalt_non_ukb_amr_XX=0;AC_non_ukb_amr_XY=0;AF_non_ukb_amr_XY=0.00000;AN_non_ukb_amr_XY=6770;nhomalt_non_ukb_amr_XY=0;AC_non_ukb_amr=0;AF_non_ukb_amr=0.00000;AN_non_ukb_amr=14100;nhomalt_non_ukb_amr=0;AC_non_ukb_asj_XX=0;AF_non_ukb_asj_XX=0.00000;AN_non_ukb_asj_XX=5350;nhomalt_non_ukb_asj_XX=0;AC_non_ukb_asj_XY=0;AF_non_ukb_asj_XY=0.00000;AN_non_ukb_asj_XY=6088;nhomalt_non_ukb_asj_XY=0;AC_non_ukb_asj=0;AF_non_ukb_asj=0.00000;AN_non_ukb_asj=11438;nhomalt_non_ukb_asj=0;AC_non_ukb_eas_XX=0;AF_non_ukb_eas_XX=0.00000;AN_non_ukb_eas_XX=11018;nhomalt_non_ukb_eas_XX=0;AC_non_ukb_eas_XY=0;AF_non_ukb_eas_XY=0.00000;AN_non_ukb_eas_XY=11160;nhomalt_non_ukb_eas_XY=0;AC_non_ukb_eas=0;AF_non_ukb_eas=0.00000;AN_non_ukb_eas=22178;nhomalt_non_ukb_eas=0;AC_non_ukb_fin_XX=0;AF_non_ukb_fin_XX=0.00000;AN_non_ukb_fin_XX=11260;nhomalt_non_ukb_fin_XX=0;AC_non_ukb_fin_XY=0;AF_non_ukb_fin_XY=0.00000;AN_non_ukb_fin_XY=10076;nhomalt_non_ukb_fin_XY=0;AC_non_ukb_fin=0;AF_non_ukb_fin=0.00000;AN_non_ukb_fin=21336;nhomalt_non_ukb_fin=0;AC_non_ukb_mid_XX=0;AF_non_ukb_mid_XX=0.00000;AN_non_ukb_mid_XX=658;nhomalt_non_ukb_mid_XX=0;AC_non_ukb_mid_XY=0;AF_non_ukb_mid_XY=0.00000;AN_non_ukb_mid_XY=906;nhomalt_non_ukb_mid_XY=0;AC_non_ukb_mid=0;AF_non_ukb_mid=0.00000;AN_non_ukb_mid=1564;nhomalt_non_ukb_mid=0;AC_non_ukb_nfe_XX=0;AF_non_ukb_nfe_XX=0.00000;AN_non_ukb_nfe_XX=106212;nhomalt_non_ukb_nfe_XX=0;AC_non_ukb_nfe_XY=0;AF_non_ukb_nfe_XY=0.00000;AN_non_ukb_nfe_XY=116692;nhomalt_non_ukb_nfe_XY=0;AC_non_ukb_nfe=0;AF_non_ukb_nfe=0.00000;AN_non_ukb_nfe=222904;nhomalt_non_ukb_nfe=0;AC_non_ukb_raw=2;AF_non_ukb_raw=3.18074e-06;AN_non_ukb_raw=628784;nhomalt_non_ukb_raw=1;AC_non_ukb_remaining_XX=0;AF_non_ukb_remaining_XX=0.00000;AN_non_ukb_remaining_XX=10542;nhomalt_non_ukb_remaining_XX=0;AC_non_ukb_remaining_XY=0;AF_non_ukb_remaining_XY=0.00000;AN_non_ukb_remaining_XY=9978;nhomalt_non_ukb_remaining_XY=0;AC_non_ukb_remaining=0;AF_non_ukb_remaining=0.00000;AN_non_ukb_remaining=20520;nhomalt_non_ukb_remaining=0;AC_non_ukb_sas_XX=0;AF_non_ukb_sas_XX=0.00000;AN_non_ukb_sas_XX=7956;nhomalt_non_ukb_sas_XX=0;AC_non_ukb_sas_XY=0;AF_non_ukb_sas_XY=0.00000;AN_non_ukb_sas_XY=31954;nhomalt_non_ukb_sas_XY=0;AC_non_ukb_sas=0;AF_non_ukb_sas=0.00000;AN_non_ukb_sas=39910;nhomalt_non_ukb_sas=0;AC_raw=3;AF_raw=2.16814e-06;AN_raw=1383676;nhomalt_raw=1;AC_remaining_XX=0;AF_remaining_XX=0.00000;AN_remaining_XX=14894;nhomalt_remaining_XX=0;AC_remaining_XY=0;AF_remaining_XY=0.00000;AN_remaining_XY=13694;nhomalt_remaining_XY=0;AC_remaining=0;AF_remaining=0.00000;AN_remaining=28588;nhomalt_remaining=0;AC_sas_XX=0;AF_sas_XX=0.00000;AN_sas_XX=10120;nhomalt_sas_XX=0;AC_sas_XY=0;AF_sas_XY=0.00000;AN_sas_XY=34820;nhomalt_sas_XY=0;AC_sas=0;AF_sas=0.00000;AN_sas=44940;nhomalt_sas=0;faf95=0.00000;faf95_afr=0.00000;faf95_amr=0.00000;faf95_eas=0.00000;faf95_mid=0.00000;faf95_nfe=0.00000;faf95_non_ukb=0.00000;faf95_non_ukb_afr=0.00000;faf95_non_ukb_amr=0.00000;faf95_non_ukb_eas=0.00000;faf95_non_ukb_mid=0.00000;faf95_non_ukb_nfe=0.00000;faf95_non_ukb_sas=0.00000;faf95_sas=0.00000;faf99=0.00000;faf99_afr=0.00000;faf99_amr=0.00000;faf99_eas=0.00000;faf99_mid=0.00000;faf99_nfe=0.00000;faf99_non_ukb=0.00000;faf99_non_ukb_afr=0.00000;faf99_non_ukb_amr=0.00000;faf99_non_ukb_eas=0.00000;faf99_non_ukb_mid=0.00000;faf99_non_ukb_nfe=0.00000;faf99_non_ukb_sas=0.00000;faf99_sas=0.00000;age_hist_het_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_het_n_smaller=0;age_hist_het_n_larger=0;age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_hom_n_smaller=0;age_hist_hom_n_larger=0;FS=10.6714;MQ=60.0000;MQRankSum=0.00000;QUALapprox=417;QD=9.26667;ReadPosRankSum=0.366000;SOR=0.0931772;VarDP=45;AS_FS=10.2874;AS_MQ=60.0000;AS_MQRankSum=0.00000;AS_pab_max=1.00000;AS_QUALapprox=102;AS_QD=20.4000;AS_ReadPosRankSum=0.00000;AS_SB_TABLE=23,4|1,2;AS_SOR=0.830037;AS_VarDP=5;inbreeding_coeff=0.666666;AS_culprit=AS_FS;AS_VQSLOD=2.95280;allele_type=snv;n_alt_alleles=2;variant_type=multi-snv;outside_ukb_capture_region;outside_broad_capture_region;gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;gq_hist_all_bin_freq=0|0|0|0|45189|0|106022|0|141915|0|8923|0|5257|0|0|0|0|0|0|0;dp_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_n_larger=0;dp_hist_all_bin_freq=0|0|129290|66843|37906|21779|16565|9557|6788|5213|3730|2737|1924|1319|897|679|510|336|233|265;dp_hist_all_n_larger=735;ab_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;cadd_raw_score=0.498516;cadd_phred=6.48900;spliceai_ds_max=0.00000;pangolin_largest_ds=0.00000;phylop=0.575000;VRS_Allele_IDs=ga4gh:VA.GgqIYByvIrViXDBdBWG4hQrcrqDJKvrX,ga4gh:VA.MgI2YPjBm3-AgSSENp7P_83rYumWa-YP;VRS_Starts=48379903,48379903;VRS_Ends=48379904,48379904;VRS_States=G,A;vep=A|intron_variant|MODIFIER|RB1|ENSG00000139687|Transcript|ENST00000267163|protein_coding||14/26|ENST00000267163.6:c.1390-149G>A|||||||1||1||SNV|HGNC|HGNC:9884|YES|NM_000321.3||1|P1|CCDS31973.1|ENSP00000267163||Ensembl|||||||||||||,A|upstream_gene_variant|MODIFIER|RB1|ENSG00000139687|Transcript|ENST00000643064|protein_coding||||||||||1|1346|1||SNV|HGNC|HGNC:9884|||||||ENSP00000496005||Ensembl|||||||||||||,A|intron_variant|MODIFIER|RB1|ENSG00000139687|Transcript|ENST00000650461|protein_coding||14/26|ENST00000650461.1:c.1390-149G>A|||||||1||1||SNV|HGNC|HGNC:9884|||||||ENSP00000497193||Ensembl|||||||||||||,A|intron_variant|MODIFIER|RB1|5925|Transcript|NM_000321.3|protein_coding||14/26|NM_000321.3:c.1390-149G>A|||||||1||1||SNV|EntrezGene|HGNC:9884|YES|ENST00000267163.6|||||NP_000312.2||RefSeq|||||||||||||,A|intron_variant|MODIFIER|RB1|5925|Transcript|XM_011535171.2|protein_coding||13/25|XM_011535171.2:c.1129-149G>A|||||||1||1||SNV|EntrezGene|HGNC:9884|||||||XP_011533473.1||RefSeq|||||||||||||,A|upstream_gene_variant|MODIFIER|LOC112268118|112268118|Transcript|XR_002957522.1|lncRNA||||||||||1|30|-1||SNV|EntrezGene||YES||||||||RefSeq|||||||||||||\n",
      "\n",
      "\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:23)\n",
      "\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:23)\n",
      "\tat is.hail.utils.package$.fatal(package.scala:96)\n",
      "\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2106)\n",
      "\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7$adapted(LoadVCF.scala:2076)\n",
      "\tat scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:515)\n",
      "\tat scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.__m633split_StreamFor_region3_18(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.__m633split_StreamFor(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.__m563split_Block(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n",
      "\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$10(BackendUtils.scala:90)\n",
      "\tat is.hail.utils.package$.using(package.scala:673)\n",
      "\tat is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:166)\n",
      "\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$9(BackendUtils.scala:89)\n",
      "\tat is.hail.backend.spark.SparkBackend$$anon$5.compute(SparkBackend.scala:444)\n",
      "\tat org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)\n",
      "\tat org.apache.spark.rdd.RDD.iterator(RDD.scala:331)\n",
      "\tat org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)\n",
      "\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)\n",
      "\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\n",
      "\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)\n",
      "\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)\n",
      "\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)\n",
      "\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)\n",
      "\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:1583)\n",
      "Caused by: is.hail.utils.HailException: Invalid locus 'chr13:48379904' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\n",
      "\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:19)\n",
      "\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:19)\n",
      "\tat is.hail.utils.package$.fatal(package.scala:96)\n",
      "\tat is.hail.variant.ReferenceGenome.checkLocus(ReferenceGenome.scala:237)\n",
      "\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3(LoadVCF.scala:521)\n",
      "\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3$adapted(LoadVCF.scala:521)\n",
      "\tat scala.Option.foreach(Option.scala:407)\n",
      "\tat is.hail.io.vcf.VCFLine.parseAddVariant(LoadVCF.scala:521)\n",
      "\tat is.hail.io.vcf.LoadVCF$.parseLine(LoadVCF.scala:1483)\n",
      "\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2092)\n",
      "\t... 26 more\n",
      "\n",
      "Driver stacktrace:\n",
      "\tat org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)\n",
      "\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)\n",
      "\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)\n",
      "\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n",
      "\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n",
      "\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n",
      "\tat org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)\n",
      "\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)\n",
      "\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)\n",
      "\tat scala.Option.foreach(Option.scala:407)\n",
      "\tat org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)\n",
      "\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)\n",
      "\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)\n",
      "\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)\n",
      "\tat org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)\n",
      "\tat org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)\n",
      "\tat org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)\n",
      "\tat is.hail.backend.spark.SparkBackend.$anonfun$parallelizeAndComputeWithIndex$4(SparkBackend.scala:455)\n",
      "\tat is.hail.backend.spark.SparkBackend.$anonfun$parallelizeAndComputeWithIndex$4$adapted(SparkBackend.scala:454)\n",
      "\tat scala.collection.Iterator.foreach(Iterator.scala:943)\n",
      "\tat scala.collection.Iterator.foreach$(Iterator.scala:943)\n",
      "\tat scala.collection.AbstractIterator.foreach(Iterator.scala:1431)\n",
      "\tat is.hail.backend.spark.SparkBackend.parallelizeAndComputeWithIndex(SparkBackend.scala:454)\n",
      "\tat is.hail.backend.BackendUtils.collectDArray(BackendUtils.scala:85)\n",
      "\tat __C540Compiled.apply(Emit.scala)\n",
      "\tat is.hail.expr.ir.LoweredTableReader$.$anonfun$makeCoercer$15(TableIR.scala:402)\n",
      "\tat is.hail.backend.ExecuteContext.$anonfun$scopedExecution$1(ExecuteContext.scala:144)\n",
      "\tat is.hail.utils.package$.using(package.scala:673)\n",
      "\tat is.hail.backend.ExecuteContext.scopedExecution(ExecuteContext.scala:144)\n",
      "\tat is.hail.expr.ir.LoweredTableReader$.makeCoercer(TableIR.scala:401)\n",
      "\tat is.hail.expr.ir.GenericTableValue.getLTVCoercer(GenericTableValue.scala:187)\n",
      "\tat is.hail.expr.ir.GenericTableValue.toTableStage(GenericTableValue.scala:220)\n",
      "\tat is.hail.io.vcf.MatrixVCFReader.lower(LoadVCF.scala:2140)\n",
      "\tat is.hail.expr.ir.TableReader.lower(TableIR.scala:610)\n",
      "\tat is.hail.expr.ir.lowering.LowerTableIR$.applyTable(LowerTableIR.scala:1062)\n",
      "\tat is.hail.expr.ir.lowering.LowerTableIR$.lower$1(LowerTableIR.scala:728)\n",
      "\tat is.hail.expr.ir.lowering.LowerTableIR$.apply(LowerTableIR.scala:732)\n",
      "\tat is.hail.expr.ir.lowering.LowerToCDA$.lower(LowerToCDA.scala:27)\n",
      "\tat is.hail.expr.ir.lowering.LowerToCDA$.apply(LowerToCDA.scala:11)\n",
      "\tat is.hail.expr.ir.lowering.LowerToDistributedArrayPass.transform(LoweringPass.scala:91)\n",
      "\tat is.hail.expr.ir.LowerOrInterpretNonCompilable$.evaluate$1(LowerOrInterpretNonCompilable.scala:27)\n",
      "\tat is.hail.expr.ir.LowerOrInterpretNonCompilable$.rewrite$1(LowerOrInterpretNonCompilable.scala:59)\n",
      "\tat is.hail.expr.ir.LowerOrInterpretNonCompilable$.apply(LowerOrInterpretNonCompilable.scala:64)\n",
      "\tat is.hail.expr.ir.lowering.LowerOrInterpretNonCompilablePass$.transform(LoweringPass.scala:83)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$3(LoweringPass.scala:32)\n",
      "\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$1(LoweringPass.scala:32)\n",
      "\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPass.apply(LoweringPass.scala:30)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPass.apply$(LoweringPass.scala:29)\n",
      "\tat is.hail.expr.ir.lowering.LowerOrInterpretNonCompilablePass$.apply(LoweringPass.scala:78)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1(LoweringPipeline.scala:21)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1$adapted(LoweringPipeline.scala:19)\n",
      "\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n",
      "\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n",
      "\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPipeline.apply(LoweringPipeline.scala:19)\n",
      "\tat is.hail.expr.ir.lowering.EvalRelationalLets$.execute$1(EvalRelationalLets.scala:13)\n",
      "\tat is.hail.expr.ir.lowering.EvalRelationalLets$.lower$1(EvalRelationalLets.scala:21)\n",
      "\tat is.hail.expr.ir.lowering.EvalRelationalLets$.apply(EvalRelationalLets.scala:35)\n",
      "\tat is.hail.expr.ir.lowering.EvalRelationalLetsPass.transform(LoweringPass.scala:168)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$3(LoweringPass.scala:32)\n",
      "\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$1(LoweringPass.scala:32)\n",
      "\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPass.apply(LoweringPass.scala:30)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPass.apply$(LoweringPass.scala:29)\n",
      "\tat is.hail.expr.ir.lowering.EvalRelationalLetsPass.apply(LoweringPass.scala:162)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1(LoweringPipeline.scala:21)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1$adapted(LoweringPipeline.scala:19)\n",
      "\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n",
      "\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n",
      "\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n",
      "\tat is.hail.expr.ir.lowering.LoweringPipeline.apply(LoweringPipeline.scala:19)\n",
      "\tat is.hail.expr.ir.CompileAndEvaluate$._apply(CompileAndEvaluate.scala:45)\n",
      "\tat is.hail.backend.spark.SparkBackend._execute(SparkBackend.scala:578)\n",
      "\tat is.hail.backend.spark.SparkBackend.$anonfun$execute$4(SparkBackend.scala:614)\n",
      "\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n",
      "\tat is.hail.backend.spark.SparkBackend.$anonfun$execute$3(SparkBackend.scala:609)\n",
      "\tat is.hail.backend.spark.SparkBackend.$anonfun$execute$3$adapted(SparkBackend.scala:608)\n",
      "\tat is.hail.backend.ExecuteContext$.$anonfun$scoped$3(ExecuteContext.scala:78)\n",
      "\tat is.hail.utils.package$.using(package.scala:673)\n",
      "\tat is.hail.backend.ExecuteContext$.$anonfun$scoped$2(ExecuteContext.scala:78)\n",
      "\tat is.hail.utils.package$.using(package.scala:673)\n",
      "\tat is.hail.annotations.RegionPool$.scoped(RegionPool.scala:13)\n",
      "\tat is.hail.backend.ExecuteContext$.scoped(ExecuteContext.scala:65)\n",
      "\tat is.hail.backend.spark.SparkBackend.$anonfun$withExecuteContext$2(SparkBackend.scala:411)\n",
      "\tat is.hail.utils.ExecutionTimer$.time(ExecutionTimer.scala:55)\n",
      "\tat is.hail.utils.ExecutionTimer$.logTime(ExecutionTimer.scala:62)\n",
      "\tat is.hail.backend.spark.SparkBackend.withExecuteContext(SparkBackend.scala:397)\n",
      "\tat is.hail.backend.spark.SparkBackend.execute(SparkBackend.scala:608)\n",
      "\tat is.hail.backend.BackendHttpHandler.handle(BackendServer.scala:88)\n",
      "\tat jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)\n",
      "\tat jdk.httpserver/sun.net.httpserver.AuthFilter.doFilter(AuthFilter.java:82)\n",
      "\tat jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:101)\n",
      "\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Exchange$LinkHandler.handle(ServerImpl.java:871)\n",
      "\tat jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)\n",
      "\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Exchange.run(ServerImpl.java:847)\n",
      "\tat jdk.httpserver/sun.net.httpserver.ServerImpl$DefaultExecutor.execute(ServerImpl.java:203)\n",
      "\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Dispatcher.handle(ServerImpl.java:566)\n",
      "\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Dispatcher.run(ServerImpl.java:531)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:1583)\n",
      "\n",
      "is.hail.utils.HailException: file:/C:/Users/jmcdonald/Desktop/Projects/The Genome Project/gnomad.exomes.v4.1.sites.chr13.vcf.bgz:offset 107083150656189: error while parsing line\n",
      "chr13\t48379904\t.\tG\tA\t.\tAC0\tAC=0;AN=614612;AF=0.00000;AC_XX=0;AF_XX=0.00000;AN_XX=297944;nhomalt_XX=0;AC_XY=0;AF_XY=0.00000;AN_XY=316668;nhomalt_XY=0;nhomalt=0;AC_afr_XX=0;AF_afr_XX=0.00000;AN_afr_XX=8146;nhomalt_afr_XX=0;AC_afr_XY=0;AF_afr_XY=0.00000;AN_afr_XY=6494;nhomalt_afr_XY=0;AC_afr=0;AF_afr=0.00000;AN_afr=14640;nhomalt_afr=0;AC_amr_XX=0;AF_amr_XX=0.00000;AN_amr_XX=7498;nhomalt_amr_XX=0;AC_amr_XY=0;AF_amr_XY=0.00000;AN_amr_XY=6868;nhomalt_amr_XY=0;AC_amr=0;AF_amr=0.00000;AN_amr=14366;nhomalt_amr=0;AC_asj_XX=0;AF_asj_XX=0.00000;AN_asj_XX=6192;nhomalt_asj_XX=0;AC_asj_XY=0;AF_asj_XY=0.00000;AN_asj_XY=6840;nhomalt_asj_XY=0;AC_asj=0;AF_asj=0.00000;AN_asj=13032;nhomalt_asj=0;AC_eas_XX=0;AF_eas_XX=0.00000;AN_eas_XX=11742;nhomalt_eas_XX=0;AC_eas_XY=0;AF_eas_XY=0.00000;AN_eas_XY=11538;nhomalt_eas_XY=0;AC_eas=0;AF_eas=0.00000;AN_eas=23280;nhomalt_eas=0;AC_fin_XX=0;AF_fin_XX=0.00000;AN_fin_XX=11334;nhomalt_fin_XX=0;AC_fin_XY=0;AF_fin_XY=0.00000;AN_fin_XY=10082;nhomalt_fin_XY=0;AC_fin=0;AF_fin=0.00000;AN_fin=21416;nhomalt_fin=0;AC_mid_XX=0;AF_mid_XX=0.00000;AN_mid_XX=814;nhomalt_mid_XX=0;AC_mid_XY=0;AF_mid_XY=0.00000;AN_mid_XY=1260;nhomalt_mid_XY=0;AC_mid=0;AF_mid=0.00000;AN_mid=2074;nhomalt_mid=0;AC_nfe_XX=0;AF_nfe_XX=0.00000;AN_nfe_XX=227204;nhomalt_nfe_XX=0;AC_nfe_XY=0;AF_nfe_XY=0.00000;AN_nfe_XY=225072;nhomalt_nfe_XY=0;AC_nfe=0;AF_nfe=0.00000;AN_nfe=452276;nhomalt_nfe=0;AC_non_ukb_XX=0;AF_non_ukb_XX=0.00000;AN_non_ukb_XX=165874;nhomalt_non_ukb_XX=0;AC_non_ukb_XY=0;AF_non_ukb_XY=0.00000;AN_non_ukb_XY=198072;nhomalt_non_ukb_XY=0;AC_non_ukb=0;AF_non_ukb=0.00000;AN_non_ukb=363946;nhomalt_non_ukb=0;AC_non_ukb_afr_XX=0;AF_non_ukb_afr_XX=0.00000;AN_non_ukb_afr_XX=5548;nhomalt_non_ukb_afr_XX=0;AC_non_ukb_afr_XY=0;AF_non_ukb_afr_XY=0.00000;AN_non_ukb_afr_XY=4448;nhomalt_non_ukb_afr_XY=0;AC_non_ukb_afr=0;AF_non_ukb_afr=0.00000;AN_non_ukb_afr=9996;nhomalt_non_ukb_afr=0;AC_non_ukb_amr_XX=0;AF_non_ukb_amr_XX=0.00000;AN_non_ukb_amr_XX=7330;nhomalt_non_ukb_amr_XX=0;AC_non_ukb_amr_XY=0;AF_non_ukb_amr_XY=0.00000;AN_non_ukb_amr_XY=6770;nhomalt_non_ukb_amr_XY=0;AC_non_ukb_amr=0;AF_non_ukb_amr=0.00000;AN_non_ukb_amr=14100;nhomalt_non_ukb_amr=0;AC_non_ukb_asj_XX=0;AF_non_ukb_asj_XX=0.00000;AN_non_ukb_asj_XX=5350;nhomalt_non_ukb_asj_XX=0;AC_non_ukb_asj_XY=0;AF_non_ukb_asj_XY=0.00000;AN_non_ukb_asj_XY=6088;nhomalt_non_ukb_asj_XY=0;AC_non_ukb_asj=0;AF_non_ukb_asj=0.00000;AN_non_ukb_asj=11438;nhomalt_non_ukb_asj=0;AC_non_ukb_eas_XX=0;AF_non_ukb_eas_XX=0.00000;AN_non_ukb_eas_XX=11018;nhomalt_non_ukb_eas_XX=0;AC_non_ukb_eas_XY=0;AF_non_ukb_eas_XY=0.00000;AN_non_ukb_eas_XY=11160;nhomalt_non_ukb_eas_XY=0;AC_non_ukb_eas=0;AF_non_ukb_eas=0.00000;AN_non_ukb_eas=22178;nhomalt_non_ukb_eas=0;AC_non_ukb_fin_XX=0;AF_non_ukb_fin_XX=0.00000;AN_non_ukb_fin_XX=11260;nhomalt_non_ukb_fin_XX=0;AC_non_ukb_fin_XY=0;AF_non_ukb_fin_XY=0.00000;AN_non_ukb_fin_XY=10076;nhomalt_non_ukb_fin_XY=0;AC_non_ukb_fin=0;AF_non_ukb_fin=0.00000;AN_non_ukb_fin=21336;nhomalt_non_ukb_fin=0;AC_non_ukb_mid_XX=0;AF_non_ukb_mid_XX=0.00000;AN_non_ukb_mid_XX=658;nhomalt_non_ukb_mid_XX=0;AC_non_ukb_mid_XY=0;AF_non_ukb_mid_XY=0.00000;AN_non_ukb_mid_XY=906;nhomalt_non_ukb_mid_XY=0;AC_non_ukb_mid=0;AF_non_ukb_mid=0.00000;AN_non_ukb_mid=1564;nhomalt_non_ukb_mid=0;AC_non_ukb_nfe_XX=0;AF_non_ukb_nfe_XX=0.00000;AN_non_ukb_nfe_XX=106212;nhomalt_non_ukb_nfe_XX=0;AC_non_ukb_nfe_XY=0;AF_non_ukb_nfe_XY=0.00000;AN_non_ukb_nfe_XY=116692;nhomalt_non_ukb_nfe_XY=0;AC_non_ukb_nfe=0;AF_non_ukb_nfe=0.00000;AN_non_ukb_nfe=222904;nhomalt_non_ukb_nfe=0;AC_non_ukb_raw=2;AF_non_ukb_raw=3.18074e-06;AN_non_ukb_raw=628784;nhomalt_non_ukb_raw=1;AC_non_ukb_remaining_XX=0;AF_non_ukb_remaining_XX=0.00000;AN_non_ukb_remaining_XX=10542;nhomalt_non_ukb_remaining_XX=0;AC_non_ukb_remaining_XY=0;AF_non_ukb_remaining_XY=0.00000;AN_non_ukb_remaining_XY=9978;nhomalt_non_ukb_remaining_XY=0;AC_non_ukb_remaining=0;AF_non_ukb_remaining=0.00000;AN_non_ukb_remaining=20520;nhomalt_non_ukb_remaining=0;AC_non_ukb_sas_XX=0;AF_non_ukb_sas_XX=0.00000;AN_non_ukb_sas_XX=7956;nhomalt_non_ukb_sas_XX=0;AC_non_ukb_sas_XY=0;AF_non_ukb_sas_XY=0.00000;AN_non_ukb_sas_XY=31954;nhomalt_non_ukb_sas_XY=0;AC_non_ukb_sas=0;AF_non_ukb_sas=0.00000;AN_non_ukb_sas=39910;nhomalt_non_ukb_sas=0;AC_raw=3;AF_raw=2.16814e-06;AN_raw=1383676;nhomalt_raw=1;AC_remaining_XX=0;AF_remaining_XX=0.00000;AN_remaining_XX=14894;nhomalt_remaining_XX=0;AC_remaining_XY=0;AF_remaining_XY=0.00000;AN_remaining_XY=13694;nhomalt_remaining_XY=0;AC_remaining=0;AF_remaining=0.00000;AN_remaining=28588;nhomalt_remaining=0;AC_sas_XX=0;AF_sas_XX=0.00000;AN_sas_XX=10120;nhomalt_sas_XX=0;AC_sas_XY=0;AF_sas_XY=0.00000;AN_sas_XY=34820;nhomalt_sas_XY=0;AC_sas=0;AF_sas=0.00000;AN_sas=44940;nhomalt_sas=0;faf95=0.00000;faf95_afr=0.00000;faf95_amr=0.00000;faf95_eas=0.00000;faf95_mid=0.00000;faf95_nfe=0.00000;faf95_non_ukb=0.00000;faf95_non_ukb_afr=0.00000;faf95_non_ukb_amr=0.00000;faf95_non_ukb_eas=0.00000;faf95_non_ukb_mid=0.00000;faf95_non_ukb_nfe=0.00000;faf95_non_ukb_sas=0.00000;faf95_sas=0.00000;faf99=0.00000;faf99_afr=0.00000;faf99_amr=0.00000;faf99_eas=0.00000;faf99_mid=0.00000;faf99_nfe=0.00000;faf99_non_ukb=0.00000;faf99_non_ukb_afr=0.00000;faf99_non_ukb_amr=0.00000;faf99_non_ukb_eas=0.00000;faf99_non_ukb_mid=0.00000;faf99_non_ukb_nfe=0.00000;faf99_non_ukb_sas=0.00000;faf99_sas=0.00000;age_hist_het_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_het_n_smaller=0;age_hist_het_n_larger=0;age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_hom_n_smaller=0;age_hist_hom_n_larger=0;FS=10.6714;MQ=60.0000;MQRankSum=0.00000;QUALapprox=417;QD=9.26667;ReadPosRankSum=0.366000;SOR=0.0931772;VarDP=45;AS_FS=10.2874;AS_MQ=60.0000;AS_MQRankSum=0.00000;AS_pab_max=1.00000;AS_QUALapprox=102;AS_QD=20.4000;AS_ReadPosRankSum=0.00000;AS_SB_TABLE=23,4|1,2;AS_SOR=0.830037;AS_VarDP=5;inbreeding_coeff=0.666666;AS_culprit=AS_FS;AS_VQSLOD=2.95280;allele_type=snv;n_alt_alleles=2;variant_type=multi-snv;outside_ukb_capture_region;outside_broad_capture_region;gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;gq_hist_all_bin_freq=0|0|0|0|45189|0|106022|0|141915|0|8923|0|5257|0|0|0|0|0|0|0;dp_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_n_larger=0;dp_hist_all_bin_freq=0|0|129290|66843|37906|21779|16565|9557|6788|5213|3730|2737|1924|1319|897|679|510|336|233|265;dp_hist_all_n_larger=735;ab_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;cadd_raw_score=0.498516;cadd_phred=6.48900;spliceai_ds_max=0.00000;pangolin_largest_ds=0.00000;phylop=0.575000;VRS_Allele_IDs=ga4gh:VA.GgqIYByvIrViXDBdBWG4hQrcrqDJKvrX,ga4gh:VA.MgI2YPjBm3-AgSSENp7P_83rYumWa-YP;VRS_Starts=48379903,48379903;VRS_Ends=48379904,48379904;VRS_States=G,A;vep=A|intron_variant|MODIFIER|RB1|ENSG00000139687|Transcript|ENST00000267163|protein_coding||14/26|ENST00000267163.6:c.1390-149G>A|||||||1||1||SNV|HGNC|HGNC:9884|YES|NM_000321.3||1|P1|CCDS31973.1|ENSP00000267163||Ensembl|||||||||||||,A|upstream_gene_variant|MODIFIER|RB1|ENSG00000139687|Transcript|ENST00000643064|protein_coding||||||||||1|1346|1||SNV|HGNC|HGNC:9884|||||||ENSP00000496005||Ensembl|||||||||||||,A|intron_variant|MODIFIER|RB1|ENSG00000139687|Transcript|ENST00000650461|protein_coding||14/26|ENST00000650461.1:c.1390-149G>A|||||||1||1||SNV|HGNC|HGNC:9884|||||||ENSP00000497193||Ensembl|||||||||||||,A|intron_variant|MODIFIER|RB1|5925|Transcript|NM_000321.3|protein_coding||14/26|NM_000321.3:c.1390-149G>A|||||||1||1||SNV|EntrezGene|HGNC:9884|YES|ENST00000267163.6|||||NP_000312.2||RefSeq|||||||||||||,A|intron_variant|MODIFIER|RB1|5925|Transcript|XM_011535171.2|protein_coding||13/25|XM_011535171.2:c.1129-149G>A|||||||1||1||SNV|EntrezGene|HGNC:9884|||||||XP_011533473.1||RefSeq|||||||||||||,A|upstream_gene_variant|MODIFIER|LOC112268118|112268118|Transcript|XR_002957522.1|lncRNA||||||||||1|30|-1||SNV|EntrezGene||YES||||||||RefSeq|||||||||||||\n",
      "\n",
      "\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:23)\n",
      "\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:23)\n",
      "\tat is.hail.utils.package$.fatal(package.scala:96)\n",
      "\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2106)\n",
      "\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7$adapted(LoadVCF.scala:2076)\n",
      "\tat scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:515)\n",
      "\tat scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.__m633split_StreamFor_region3_18(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.__m633split_StreamFor(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.__m563split_Block(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n",
      "\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$10(BackendUtils.scala:90)\n",
      "\tat is.hail.utils.package$.using(package.scala:673)\n",
      "\tat is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:166)\n",
      "\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$9(BackendUtils.scala:89)\n",
      "\tat is.hail.backend.spark.SparkBackend$$anon$5.compute(SparkBackend.scala:444)\n",
      "\tat org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)\n",
      "\tat org.apache.spark.rdd.RDD.iterator(RDD.scala:331)\n",
      "\tat org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)\n",
      "\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)\n",
      "\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\n",
      "\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)\n",
      "\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)\n",
      "\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)\n",
      "\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)\n",
      "\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:1583)\n",
      "\n",
      "is.hail.utils.HailException: Invalid locus 'chr13:48379904' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\n",
      "\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:19)\n",
      "\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:19)\n",
      "\tat is.hail.utils.package$.fatal(package.scala:96)\n",
      "\tat is.hail.variant.ReferenceGenome.checkLocus(ReferenceGenome.scala:237)\n",
      "\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3(LoadVCF.scala:521)\n",
      "\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3$adapted(LoadVCF.scala:521)\n",
      "\tat scala.Option.foreach(Option.scala:407)\n",
      "\tat is.hail.io.vcf.VCFLine.parseAddVariant(LoadVCF.scala:521)\n",
      "\tat is.hail.io.vcf.LoadVCF$.parseLine(LoadVCF.scala:1483)\n",
      "\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2092)\n",
      "\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7$adapted(LoadVCF.scala:2076)\n",
      "\tat scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:515)\n",
      "\tat scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.__m633split_StreamFor_region3_18(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.__m633split_StreamFor(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.__m563split_Block(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n",
      "\tat __C551collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n",
      "\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$10(BackendUtils.scala:90)\n",
      "\tat is.hail.utils.package$.using(package.scala:673)\n",
      "\tat is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:166)\n",
      "\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$9(BackendUtils.scala:89)\n",
      "\tat is.hail.backend.spark.SparkBackend$$anon$5.compute(SparkBackend.scala:444)\n",
      "\tat org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)\n",
      "\tat org.apache.spark.rdd.RDD.iterator(RDD.scala:331)\n",
      "\tat org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)\n",
      "\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)\n",
      "\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\n",
      "\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)\n",
      "\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)\n",
      "\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)\n",
      "\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)\n",
      "\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n",
      "\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n",
      "\tat java.base/java.lang.Thread.run(Thread.java:1583)\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Hail version: 0.2.133-4c60fddb171a\n",
      "Error summary: HailException: Invalid locus 'chr13:48379904' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\n",
      "Number of samples: 0\n"
     ]
    }
   ],
   "source": [
    "# Configure your VCF file path\n",
    "vcf_path = r'C:\\Users\\jmcdonald\\Desktop\\Projects\\The Genome Project\\gnomad.exomes.v4.1.sites.chr13.vcf.bgz'\n",
    "print(f\"Checking if file exists: {os.path.exists(vcf_path)}\")\n",
    "log_step(\"Loading VCF file...\")\n",
    "try:\n",
    "    mt = hl.import_vcf(vcf_path)\n",
    "    \n",
    "    # Try getting the counts in separate try-except blocks\n",
    "    try:\n",
    "        n_variants = mt.count_rows()\n",
    "        print(f\"Number of variants: {n_variants:,}\")\n",
    "    except Exception as e:\n",
    "        print(f\"Error counting variants: {e}\")\n",
    "    \n",
    "    try:\n",
    "        n_samples = mt.count_cols()\n",
    "        print(f\"Number of samples: {n_samples:,}\")\n",
    "    except Exception as e:\n",
    "        print(f\"Error counting samples: {e}\")\n",
    "        \n",
    "except Exception as e:\n",
    "    print(f\"Error loading VCF: {e}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initial dataset dimensions:\n"
     ]
    },
    {
     "ename": "FatalError",
     "evalue": "HailException: Invalid locus 'chr13:108215366' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\n\nJava stack trace:\norg.apache.spark.SparkException: Job aborted due to stage failure: Task 22 in stage 0.0 failed 1 times, most recent failure: Lost task 22.0 in stage 0.0 (TID 22) (LT2058-20241028.SHTC.com executor driver): is.hail.utils.HailException: file:/C:/Users/jmcdonald/Desktop/Projects/The Genome Project/gnomad.exomes.v4.1.sites.chr13.vcf.bgz:offset 196318482477539: error while parsing line\nchr13\t108215366\t.\tG\tT\t.\tAC0\tAC=0;AN=334;AF=0.00000;AC_XX=0;AF_XX=0.00000;AN_XX=88;nhomalt_XX=0;AC_XY=0;AF_XY=0.00000;AN_XY=246;nhomalt_XY=0;nhomalt=0;AC_afr_XX=0;AF_afr_XX=0.00000;AN_afr_XX=4;nhomalt_afr_XX=0;AC_afr_XY=0;AF_afr_XY=0.00000;AN_afr_XY=6;nhomalt_afr_XY=0;AC_afr=0;AF_afr=0.00000;AN_afr=10;nhomalt_afr=0;AC_amr_XX=0;AN_amr_XX=0;nhomalt_amr_XX=0;AC_amr_XY=0;AF_amr_XY=0.00000;AN_amr_XY=2;nhomalt_amr_XY=0;AC_amr=0;AF_amr=0.00000;AN_amr=2;nhomalt_amr=0;AC_asj_XX=0;AN_asj_XX=0;nhomalt_asj_XX=0;AC_asj_XY=0;AF_asj_XY=0.00000;AN_asj_XY=2;nhomalt_asj_XY=0;AC_asj=0;AF_asj=0.00000;AN_asj=2;nhomalt_asj=0;AC_eas_XX=0;AN_eas_XX=0;nhomalt_eas_XX=0;AC_eas_XY=0;AF_eas_XY=0.00000;AN_eas_XY=14;nhomalt_eas_XY=0;AC_eas=0;AF_eas=0.00000;AN_eas=14;nhomalt_eas=0;AC_fin_XX=0;AN_fin_XX=0;nhomalt_fin_XX=0;AC_fin_XY=0;AN_fin_XY=0;nhomalt_fin_XY=0;AC_fin=0;AN_fin=0;nhomalt_fin=0;AC_mid_XX=0;AN_mid_XX=0;nhomalt_mid_XX=0;AC_mid_XY=0;AN_mid_XY=0;nhomalt_mid_XY=0;AC_mid=0;AN_mid=0;nhomalt_mid=0;AC_nfe_XX=0;AF_nfe_XX=0.00000;AN_nfe_XX=74;nhomalt_nfe_XX=0;AC_nfe_XY=0;AF_nfe_XY=0.00000;AN_nfe_XY=204;nhomalt_nfe_XY=0;AC_nfe=0;AF_nfe=0.00000;AN_nfe=278;nhomalt_nfe=0;AC_non_ukb_XX=0;AF_non_ukb_XX=0.00000;AN_non_ukb_XX=88;nhomalt_non_ukb_XX=0;AC_non_ukb_XY=0;AF_non_ukb_XY=0.00000;AN_non_ukb_XY=246;nhomalt_non_ukb_XY=0;AC_non_ukb=0;AF_non_ukb=0.00000;AN_non_ukb=334;nhomalt_non_ukb=0;AC_non_ukb_afr_XX=0;AF_non_ukb_afr_XX=0.00000;AN_non_ukb_afr_XX=4;nhomalt_non_ukb_afr_XX=0;AC_non_ukb_afr_XY=0;AF_non_ukb_afr_XY=0.00000;AN_non_ukb_afr_XY=6;nhomalt_non_ukb_afr_XY=0;AC_non_ukb_afr=0;AF_non_ukb_afr=0.00000;AN_non_ukb_afr=10;nhomalt_non_ukb_afr=0;AC_non_ukb_amr_XX=0;AN_non_ukb_amr_XX=0;nhomalt_non_ukb_amr_XX=0;AC_non_ukb_amr_XY=0;AF_non_ukb_amr_XY=0.00000;AN_non_ukb_amr_XY=2;nhomalt_non_ukb_amr_XY=0;AC_non_ukb_amr=0;AF_non_ukb_amr=0.00000;AN_non_ukb_amr=2;nhomalt_non_ukb_amr=0;AC_non_ukb_asj_XX=0;AN_non_ukb_asj_XX=0;nhomalt_non_ukb_asj_XX=0;AC_non_ukb_asj_XY=0;AF_non_ukb_asj_XY=0.00000;AN_non_ukb_asj_XY=2;nhomalt_non_ukb_asj_XY=0;AC_non_ukb_asj=0;AF_non_ukb_asj=0.00000;AN_non_ukb_asj=2;nhomalt_non_ukb_asj=0;AC_non_ukb_eas_XX=0;AN_non_ukb_eas_XX=0;nhomalt_non_ukb_eas_XX=0;AC_non_ukb_eas_XY=0;AF_non_ukb_eas_XY=0.00000;AN_non_ukb_eas_XY=14;nhomalt_non_ukb_eas_XY=0;AC_non_ukb_eas=0;AF_non_ukb_eas=0.00000;AN_non_ukb_eas=14;nhomalt_non_ukb_eas=0;AC_non_ukb_fin_XX=0;AN_non_ukb_fin_XX=0;nhomalt_non_ukb_fin_XX=0;AC_non_ukb_fin_XY=0;AN_non_ukb_fin_XY=0;nhomalt_non_ukb_fin_XY=0;AC_non_ukb_fin=0;AN_non_ukb_fin=0;nhomalt_non_ukb_fin=0;AC_non_ukb_mid_XX=0;AN_non_ukb_mid_XX=0;nhomalt_non_ukb_mid_XX=0;AC_non_ukb_mid_XY=0;AN_non_ukb_mid_XY=0;nhomalt_non_ukb_mid_XY=0;AC_non_ukb_mid=0;AN_non_ukb_mid=0;nhomalt_non_ukb_mid=0;AC_non_ukb_nfe_XX=0;AF_non_ukb_nfe_XX=0.00000;AN_non_ukb_nfe_XX=74;nhomalt_non_ukb_nfe_XX=0;AC_non_ukb_nfe_XY=0;AF_non_ukb_nfe_XY=0.00000;AN_non_ukb_nfe_XY=204;nhomalt_non_ukb_nfe_XY=0;AC_non_ukb_nfe=0;AF_non_ukb_nfe=0.00000;AN_non_ukb_nfe=278;nhomalt_non_ukb_nfe=0;AC_non_ukb_raw=5;AF_non_ukb_raw=7.95186e-06;AN_non_ukb_raw=628784;nhomalt_non_ukb_raw=2;AC_non_ukb_remaining_XX=0;AF_non_ukb_remaining_XX=0.00000;AN_non_ukb_remaining_XX=8;nhomalt_non_ukb_remaining_XX=0;AC_non_ukb_remaining_XY=0;AF_non_ukb_remaining_XY=0.00000;AN_non_ukb_remaining_XY=12;nhomalt_non_ukb_remaining_XY=0;AC_non_ukb_remaining=0;AF_non_ukb_remaining=0.00000;AN_non_ukb_remaining=20;nhomalt_non_ukb_remaining=0;AC_non_ukb_sas_XX=0;AF_non_ukb_sas_XX=0.00000;AN_non_ukb_sas_XX=2;nhomalt_non_ukb_sas_XX=0;AC_non_ukb_sas_XY=0;AF_non_ukb_sas_XY=0.00000;AN_non_ukb_sas_XY=6;nhomalt_non_ukb_sas_XY=0;AC_non_ukb_sas=0;AF_non_ukb_sas=0.00000;AN_non_ukb_sas=8;nhomalt_non_ukb_sas=0;AC_raw=5;AF_raw=7.94690e-06;AN_raw=629176;nhomalt_raw=2;AC_remaining_XX=0;AF_remaining_XX=0.00000;AN_remaining_XX=8;nhomalt_remaining_XX=0;AC_remaining_XY=0;AF_remaining_XY=0.00000;AN_remaining_XY=12;nhomalt_remaining_XY=0;AC_remaining=0;AF_remaining=0.00000;AN_remaining=20;nhomalt_remaining=0;AC_sas_XX=0;AF_sas_XX=0.00000;AN_sas_XX=2;nhomalt_sas_XX=0;AC_sas_XY=0;AF_sas_XY=0.00000;AN_sas_XY=6;nhomalt_sas_XY=0;AC_sas=0;AF_sas=0.00000;AN_sas=8;nhomalt_sas=0;faf95=0.00000;faf95_afr=0.00000;faf95_amr=0.00000;faf95_eas=0.00000;faf95_mid=0.00000;faf95_nfe=0.00000;faf95_non_ukb=0.00000;faf95_non_ukb_afr=0.00000;faf95_non_ukb_amr=0.00000;faf95_non_ukb_eas=0.00000;faf95_non_ukb_mid=0.00000;faf95_non_ukb_nfe=0.00000;faf95_non_ukb_sas=0.00000;faf95_sas=0.00000;faf99=0.00000;faf99_afr=0.00000;faf99_amr=0.00000;faf99_eas=0.00000;faf99_mid=0.00000;faf99_nfe=0.00000;faf99_non_ukb=0.00000;faf99_non_ukb_afr=0.00000;faf99_non_ukb_amr=0.00000;faf99_non_ukb_eas=0.00000;faf99_non_ukb_mid=0.00000;faf99_non_ukb_nfe=0.00000;faf99_non_ukb_sas=0.00000;faf99_sas=0.00000;age_hist_het_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_het_n_smaller=0;age_hist_het_n_larger=0;age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_hom_n_smaller=0;age_hist_hom_n_larger=0;FS=0.00000;MQ=57.9124;MQRankSum=0.00000;QUALapprox=221;QD=20.0909;ReadPosRankSum=0.00000;SOR=0.223144;VarDP=11;AS_FS=0.00000;AS_MQ=60.0000;AS_MQRankSum=0.00000;AS_pab_max=1.00000;AS_QUALapprox=138;AS_QD=19.7143;AS_ReadPosRankSum=0.00000;AS_SB_TABLE=0,1|3,3;AS_SOR=0.223144;AS_VarDP=7;inbreeding_coeff=0.799998;AS_culprit=AS_MQ;AS_VQSLOD=6.28100;negative_train_site;allele_type=snv;n_alt_alleles=2;variant_type=multi-snv;outside_ukb_capture_region;outside_broad_capture_region;gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;gq_hist_all_bin_freq=0|0|0|0|81|0|79|0|7|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_n_larger=0;dp_hist_all_bin_freq=0|0|160|7|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_all_n_larger=0;ab_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;cadd_raw_score=0.473593;cadd_phred=6.22600;spliceai_ds_max=0.00000;pangolin_largest_ds=0.00000;phylop=-0.280000;VRS_Allele_IDs=ga4gh:VA.Atzz2sF20yoKJ1nmL2n_X7Jjmx89Ln5i,ga4gh:VA.xeYLnnwnPXgVc24xU9lxWxWo6MYNcV2A;VRS_Starts=108215365,108215365;VRS_Ends=108215366,108215366;VRS_States=G,T;vep=T|upstream_gene_variant|MODIFIER|ABHD13|ENSG00000139826|Transcript|ENST00000375898|protein_coding||||||||||1|3026|1||SNV|HGNC|HGNC:20293|YES|NM_032859.3||1|P1|CCDS32007.1|ENSP00000365063||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000405925|protein_coding||1/1|ENST00000405925.2:c.-29+2796C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||1|P1|CCDS9508.1|ENSP00000385955||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000442234|protein_coding||1/2|ENST00000442234.6:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|YES|NM_206937.2||1|P1|CCDS9508.1|ENSP00000402030||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000611712|protein_coding||1/2|ENST00000611712.4:c.-299+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||4|P1|CCDS9508.1|ENSP00000484288||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000614526|protein_coding||1/2|ENST00000614526.2:c.-223+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||2||CCDS81779.1|ENSP00000480814||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000685338|protein_coding||1/2|ENST00000685338.1:c.-275+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000510567||Ensembl|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686095|protein_coding||||||||||1|180|-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509942||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686204|protein_coding||1/2|ENST00000686204.1:c.-186+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509685||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686913|protein_coding|1/3||ENST00000686913.1:c.-341C>A||2|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509299||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686926|protein_coding||1/1|ENST00000686926.1:c.-29+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509122||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000687164|protein_coding|1/3||ENST00000687164.1:c.-144C>A||44|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000508512||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000687822|protein_coding||1/2|ENST00000687822.1:c.-210+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509344||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688396|protein_coding||2/3|ENST00000688396.1:c.-210+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509564||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688455|protein_coding||2/3|ENST00000688455.1:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509304||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688529|protein_coding|1/3||ENST00000688529.1:c.-252C>A||36|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509906||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688595|protein_coding||2/3|ENST00000688595.1:c.-186+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509502||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000689762|protein_coding||1/2|ENST00000689762.1:c.-331+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||||CCDS81779.1|ENSP00000508867||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000690127|protein_coding|1/3||ENST00000690127.1:c.-373C>A||22|||||1||-1||SNV|HGNC|HGNC:6601||||||CCDS81779.1|ENSP00000509468||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000692222|protein_coding||1/2|ENST00000692222.1:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509226||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000693040|protein_coding||1/2|ENST00000693040.1:c.-78+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000510014||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001098268.2|protein_coding||1/1|NM_001098268.2:c.-29+2796C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001091738.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001330595.2|protein_coding||1/2|NM_001330595.2:c.-223+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001317524.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352598.2|protein_coding||2/3|NM_001352598.2:c.-186+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339527.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352599.2|protein_coding||2/3|NM_001352599.2:c.-210+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339528.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352600.2|protein_coding||2/3|NM_001352600.2:c.-102+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339529.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352601.2|protein_coding|1/3||NM_001352601.2:c.-252C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339530.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352602.2|protein_coding|1/3||NM_001352602.2:c.-144C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339531.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|3981|Transcript|NM_001352603.1|protein_coding||||||||||1|61|-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339532.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352604.2|protein_coding|1/3||NM_001352604.2:c.-136C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339533.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001379095.1|protein_coding||1/2|NM_001379095.1:c.-299+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001366024.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|3981|Transcript|NM_002312.3|protein_coding||||||||||1|584|-1||SNV|EntrezGene|HGNC:6601|||||||NP_002303.2||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|ABHD13|84945|Transcript|NM_032859.3|protein_coding||||||||||1|3026|1||SNV|EntrezGene|HGNC:20293|YES|ENST00000375898.4|||||NP_116248.2||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_206937.2|protein_coding||1/2|NM_206937.2:c.-102+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|YES|ENST00000442234.6|||||NP_996820.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|ABHD13|84945|Transcript|XM_011521128.3|protein_coding||||||||||1|3380|1||SNV|EntrezGene|HGNC:20293|||||||XP_011519430.1||RefSeq|||||||||||||,T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000273230|promoter||||||||||1||||SNV||||||||||||||||||||||||,T|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00000634950|||||||||||1||1||SNV||||||||||||||||ENSPFM0617|3|Y|-0.094|ZBTB7B&ZBTB7A&ZBTB7C||||\n\r\n\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:23)\r\n\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:23)\r\n\tat is.hail.utils.package$.fatal(package.scala:96)\r\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2106)\r\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7$adapted(LoadVCF.scala:2076)\r\n\tat scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:515)\r\n\tat scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor_region3_18(Unknown Source)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor(Unknown Source)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m64split_Block(Unknown Source)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\r\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$10(BackendUtils.scala:90)\r\n\tat is.hail.utils.package$.using(package.scala:673)\r\n\tat is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:166)\r\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$9(BackendUtils.scala:89)\r\n\tat is.hail.backend.spark.SparkBackend$$anon$5.compute(SparkBackend.scala:444)\r\n\tat org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)\r\n\tat org.apache.spark.rdd.RDD.iterator(RDD.scala:331)\r\n\tat org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)\r\n\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)\r\n\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\r\n\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)\r\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)\r\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)\r\n\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)\r\n\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\r\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\r\nCaused by: is.hail.utils.HailException: Invalid locus 'chr13:108215366' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\r\n\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:19)\r\n\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:19)\r\n\tat is.hail.utils.package$.fatal(package.scala:96)\r\n\tat is.hail.variant.ReferenceGenome.checkLocus(ReferenceGenome.scala:237)\r\n\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3(LoadVCF.scala:521)\r\n\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3$adapted(LoadVCF.scala:521)\r\n\tat scala.Option.foreach(Option.scala:407)\r\n\tat is.hail.io.vcf.VCFLine.parseAddVariant(LoadVCF.scala:521)\r\n\tat is.hail.io.vcf.LoadVCF$.parseLine(LoadVCF.scala:1483)\r\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2092)\r\n\t... 26 more\r\n\nDriver stacktrace:\n\tat org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)\n\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n\tat org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)\n\tat scala.Option.foreach(Option.scala:407)\n\tat org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)\n\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)\n\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)\n\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)\n\tat org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)\n\tat org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)\n\tat org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$parallelizeAndComputeWithIndex$4(SparkBackend.scala:455)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$parallelizeAndComputeWithIndex$4$adapted(SparkBackend.scala:454)\n\tat scala.collection.Iterator.foreach(Iterator.scala:943)\n\tat scala.collection.Iterator.foreach$(Iterator.scala:943)\n\tat scala.collection.AbstractIterator.foreach(Iterator.scala:1431)\n\tat is.hail.backend.spark.SparkBackend.parallelizeAndComputeWithIndex(SparkBackend.scala:454)\n\tat is.hail.backend.BackendUtils.collectDArray(BackendUtils.scala:85)\n\tat __C41Compiled.apply(Emit.scala)\n\tat is.hail.expr.ir.LoweredTableReader$.$anonfun$makeCoercer$15(TableIR.scala:402)\n\tat is.hail.backend.ExecuteContext.$anonfun$scopedExecution$1(ExecuteContext.scala:144)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.backend.ExecuteContext.scopedExecution(ExecuteContext.scala:144)\n\tat is.hail.expr.ir.LoweredTableReader$.makeCoercer(TableIR.scala:401)\n\tat is.hail.expr.ir.GenericTableValue.getLTVCoercer(GenericTableValue.scala:187)\n\tat is.hail.expr.ir.GenericTableValue.toTableStage(GenericTableValue.scala:220)\n\tat is.hail.io.vcf.MatrixVCFReader.lower(LoadVCF.scala:2140)\n\tat is.hail.expr.ir.TableReader.lower(TableIR.scala:610)\n\tat is.hail.expr.ir.lowering.LowerTableIR$.applyTable(LowerTableIR.scala:1062)\n\tat is.hail.expr.ir.lowering.LowerTableIR$.lower$1(LowerTableIR.scala:728)\n\tat is.hail.expr.ir.lowering.LowerTableIR$.apply(LowerTableIR.scala:732)\n\tat is.hail.expr.ir.lowering.LowerToCDA$.lower(LowerToCDA.scala:27)\n\tat is.hail.expr.ir.lowering.LowerToCDA$.apply(LowerToCDA.scala:11)\n\tat is.hail.expr.ir.lowering.LowerToDistributedArrayPass.transform(LoweringPass.scala:91)\n\tat is.hail.expr.ir.LowerOrInterpretNonCompilable$.evaluate$1(LowerOrInterpretNonCompilable.scala:27)\n\tat is.hail.expr.ir.LowerOrInterpretNonCompilable$.rewrite$1(LowerOrInterpretNonCompilable.scala:59)\n\tat is.hail.expr.ir.LowerOrInterpretNonCompilable$.apply(LowerOrInterpretNonCompilable.scala:64)\n\tat is.hail.expr.ir.lowering.LowerOrInterpretNonCompilablePass$.transform(LoweringPass.scala:83)\n\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$3(LoweringPass.scala:32)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$1(LoweringPass.scala:32)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.expr.ir.lowering.LoweringPass.apply(LoweringPass.scala:30)\n\tat is.hail.expr.ir.lowering.LoweringPass.apply$(LoweringPass.scala:29)\n\tat is.hail.expr.ir.lowering.LowerOrInterpretNonCompilablePass$.apply(LoweringPass.scala:78)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1(LoweringPipeline.scala:21)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1$adapted(LoweringPipeline.scala:19)\n\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.apply(LoweringPipeline.scala:19)\n\tat is.hail.expr.ir.lowering.EvalRelationalLets$.execute$1(EvalRelationalLets.scala:13)\n\tat is.hail.expr.ir.lowering.EvalRelationalLets$.lower$1(EvalRelationalLets.scala:21)\n\tat is.hail.expr.ir.lowering.EvalRelationalLets$.apply(EvalRelationalLets.scala:35)\n\tat is.hail.expr.ir.lowering.EvalRelationalLetsPass.transform(LoweringPass.scala:168)\n\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$3(LoweringPass.scala:32)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$1(LoweringPass.scala:32)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.expr.ir.lowering.LoweringPass.apply(LoweringPass.scala:30)\n\tat is.hail.expr.ir.lowering.LoweringPass.apply$(LoweringPass.scala:29)\n\tat is.hail.expr.ir.lowering.EvalRelationalLetsPass.apply(LoweringPass.scala:162)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1(LoweringPipeline.scala:21)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1$adapted(LoweringPipeline.scala:19)\n\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.apply(LoweringPipeline.scala:19)\n\tat is.hail.expr.ir.CompileAndEvaluate$._apply(CompileAndEvaluate.scala:45)\n\tat is.hail.backend.spark.SparkBackend._execute(SparkBackend.scala:578)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$execute$4(SparkBackend.scala:614)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$execute$3(SparkBackend.scala:609)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$execute$3$adapted(SparkBackend.scala:608)\n\tat is.hail.backend.ExecuteContext$.$anonfun$scoped$3(ExecuteContext.scala:78)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.backend.ExecuteContext$.$anonfun$scoped$2(ExecuteContext.scala:78)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.annotations.RegionPool$.scoped(RegionPool.scala:13)\n\tat is.hail.backend.ExecuteContext$.scoped(ExecuteContext.scala:65)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$withExecuteContext$2(SparkBackend.scala:411)\n\tat is.hail.utils.ExecutionTimer$.time(ExecutionTimer.scala:55)\n\tat is.hail.utils.ExecutionTimer$.logTime(ExecutionTimer.scala:62)\n\tat is.hail.backend.spark.SparkBackend.withExecuteContext(SparkBackend.scala:397)\n\tat is.hail.backend.spark.SparkBackend.execute(SparkBackend.scala:608)\n\tat is.hail.backend.BackendHttpHandler.handle(BackendServer.scala:88)\n\tat jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)\n\tat jdk.httpserver/sun.net.httpserver.AuthFilter.doFilter(AuthFilter.java:82)\n\tat jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:101)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Exchange$LinkHandler.handle(ServerImpl.java:871)\n\tat jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Exchange.run(ServerImpl.java:847)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$DefaultExecutor.execute(ServerImpl.java:203)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Dispatcher.handle(ServerImpl.java:566)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Dispatcher.run(ServerImpl.java:531)\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\n\nis.hail.utils.HailException: file:/C:/Users/jmcdonald/Desktop/Projects/The Genome Project/gnomad.exomes.v4.1.sites.chr13.vcf.bgz:offset 196318482477539: error while parsing line\nchr13\t108215366\t.\tG\tT\t.\tAC0\tAC=0;AN=334;AF=0.00000;AC_XX=0;AF_XX=0.00000;AN_XX=88;nhomalt_XX=0;AC_XY=0;AF_XY=0.00000;AN_XY=246;nhomalt_XY=0;nhomalt=0;AC_afr_XX=0;AF_afr_XX=0.00000;AN_afr_XX=4;nhomalt_afr_XX=0;AC_afr_XY=0;AF_afr_XY=0.00000;AN_afr_XY=6;nhomalt_afr_XY=0;AC_afr=0;AF_afr=0.00000;AN_afr=10;nhomalt_afr=0;AC_amr_XX=0;AN_amr_XX=0;nhomalt_amr_XX=0;AC_amr_XY=0;AF_amr_XY=0.00000;AN_amr_XY=2;nhomalt_amr_XY=0;AC_amr=0;AF_amr=0.00000;AN_amr=2;nhomalt_amr=0;AC_asj_XX=0;AN_asj_XX=0;nhomalt_asj_XX=0;AC_asj_XY=0;AF_asj_XY=0.00000;AN_asj_XY=2;nhomalt_asj_XY=0;AC_asj=0;AF_asj=0.00000;AN_asj=2;nhomalt_asj=0;AC_eas_XX=0;AN_eas_XX=0;nhomalt_eas_XX=0;AC_eas_XY=0;AF_eas_XY=0.00000;AN_eas_XY=14;nhomalt_eas_XY=0;AC_eas=0;AF_eas=0.00000;AN_eas=14;nhomalt_eas=0;AC_fin_XX=0;AN_fin_XX=0;nhomalt_fin_XX=0;AC_fin_XY=0;AN_fin_XY=0;nhomalt_fin_XY=0;AC_fin=0;AN_fin=0;nhomalt_fin=0;AC_mid_XX=0;AN_mid_XX=0;nhomalt_mid_XX=0;AC_mid_XY=0;AN_mid_XY=0;nhomalt_mid_XY=0;AC_mid=0;AN_mid=0;nhomalt_mid=0;AC_nfe_XX=0;AF_nfe_XX=0.00000;AN_nfe_XX=74;nhomalt_nfe_XX=0;AC_nfe_XY=0;AF_nfe_XY=0.00000;AN_nfe_XY=204;nhomalt_nfe_XY=0;AC_nfe=0;AF_nfe=0.00000;AN_nfe=278;nhomalt_nfe=0;AC_non_ukb_XX=0;AF_non_ukb_XX=0.00000;AN_non_ukb_XX=88;nhomalt_non_ukb_XX=0;AC_non_ukb_XY=0;AF_non_ukb_XY=0.00000;AN_non_ukb_XY=246;nhomalt_non_ukb_XY=0;AC_non_ukb=0;AF_non_ukb=0.00000;AN_non_ukb=334;nhomalt_non_ukb=0;AC_non_ukb_afr_XX=0;AF_non_ukb_afr_XX=0.00000;AN_non_ukb_afr_XX=4;nhomalt_non_ukb_afr_XX=0;AC_non_ukb_afr_XY=0;AF_non_ukb_afr_XY=0.00000;AN_non_ukb_afr_XY=6;nhomalt_non_ukb_afr_XY=0;AC_non_ukb_afr=0;AF_non_ukb_afr=0.00000;AN_non_ukb_afr=10;nhomalt_non_ukb_afr=0;AC_non_ukb_amr_XX=0;AN_non_ukb_amr_XX=0;nhomalt_non_ukb_amr_XX=0;AC_non_ukb_amr_XY=0;AF_non_ukb_amr_XY=0.00000;AN_non_ukb_amr_XY=2;nhomalt_non_ukb_amr_XY=0;AC_non_ukb_amr=0;AF_non_ukb_amr=0.00000;AN_non_ukb_amr=2;nhomalt_non_ukb_amr=0;AC_non_ukb_asj_XX=0;AN_non_ukb_asj_XX=0;nhomalt_non_ukb_asj_XX=0;AC_non_ukb_asj_XY=0;AF_non_ukb_asj_XY=0.00000;AN_non_ukb_asj_XY=2;nhomalt_non_ukb_asj_XY=0;AC_non_ukb_asj=0;AF_non_ukb_asj=0.00000;AN_non_ukb_asj=2;nhomalt_non_ukb_asj=0;AC_non_ukb_eas_XX=0;AN_non_ukb_eas_XX=0;nhomalt_non_ukb_eas_XX=0;AC_non_ukb_eas_XY=0;AF_non_ukb_eas_XY=0.00000;AN_non_ukb_eas_XY=14;nhomalt_non_ukb_eas_XY=0;AC_non_ukb_eas=0;AF_non_ukb_eas=0.00000;AN_non_ukb_eas=14;nhomalt_non_ukb_eas=0;AC_non_ukb_fin_XX=0;AN_non_ukb_fin_XX=0;nhomalt_non_ukb_fin_XX=0;AC_non_ukb_fin_XY=0;AN_non_ukb_fin_XY=0;nhomalt_non_ukb_fin_XY=0;AC_non_ukb_fin=0;AN_non_ukb_fin=0;nhomalt_non_ukb_fin=0;AC_non_ukb_mid_XX=0;AN_non_ukb_mid_XX=0;nhomalt_non_ukb_mid_XX=0;AC_non_ukb_mid_XY=0;AN_non_ukb_mid_XY=0;nhomalt_non_ukb_mid_XY=0;AC_non_ukb_mid=0;AN_non_ukb_mid=0;nhomalt_non_ukb_mid=0;AC_non_ukb_nfe_XX=0;AF_non_ukb_nfe_XX=0.00000;AN_non_ukb_nfe_XX=74;nhomalt_non_ukb_nfe_XX=0;AC_non_ukb_nfe_XY=0;AF_non_ukb_nfe_XY=0.00000;AN_non_ukb_nfe_XY=204;nhomalt_non_ukb_nfe_XY=0;AC_non_ukb_nfe=0;AF_non_ukb_nfe=0.00000;AN_non_ukb_nfe=278;nhomalt_non_ukb_nfe=0;AC_non_ukb_raw=5;AF_non_ukb_raw=7.95186e-06;AN_non_ukb_raw=628784;nhomalt_non_ukb_raw=2;AC_non_ukb_remaining_XX=0;AF_non_ukb_remaining_XX=0.00000;AN_non_ukb_remaining_XX=8;nhomalt_non_ukb_remaining_XX=0;AC_non_ukb_remaining_XY=0;AF_non_ukb_remaining_XY=0.00000;AN_non_ukb_remaining_XY=12;nhomalt_non_ukb_remaining_XY=0;AC_non_ukb_remaining=0;AF_non_ukb_remaining=0.00000;AN_non_ukb_remaining=20;nhomalt_non_ukb_remaining=0;AC_non_ukb_sas_XX=0;AF_non_ukb_sas_XX=0.00000;AN_non_ukb_sas_XX=2;nhomalt_non_ukb_sas_XX=0;AC_non_ukb_sas_XY=0;AF_non_ukb_sas_XY=0.00000;AN_non_ukb_sas_XY=6;nhomalt_non_ukb_sas_XY=0;AC_non_ukb_sas=0;AF_non_ukb_sas=0.00000;AN_non_ukb_sas=8;nhomalt_non_ukb_sas=0;AC_raw=5;AF_raw=7.94690e-06;AN_raw=629176;nhomalt_raw=2;AC_remaining_XX=0;AF_remaining_XX=0.00000;AN_remaining_XX=8;nhomalt_remaining_XX=0;AC_remaining_XY=0;AF_remaining_XY=0.00000;AN_remaining_XY=12;nhomalt_remaining_XY=0;AC_remaining=0;AF_remaining=0.00000;AN_remaining=20;nhomalt_remaining=0;AC_sas_XX=0;AF_sas_XX=0.00000;AN_sas_XX=2;nhomalt_sas_XX=0;AC_sas_XY=0;AF_sas_XY=0.00000;AN_sas_XY=6;nhomalt_sas_XY=0;AC_sas=0;AF_sas=0.00000;AN_sas=8;nhomalt_sas=0;faf95=0.00000;faf95_afr=0.00000;faf95_amr=0.00000;faf95_eas=0.00000;faf95_mid=0.00000;faf95_nfe=0.00000;faf95_non_ukb=0.00000;faf95_non_ukb_afr=0.00000;faf95_non_ukb_amr=0.00000;faf95_non_ukb_eas=0.00000;faf95_non_ukb_mid=0.00000;faf95_non_ukb_nfe=0.00000;faf95_non_ukb_sas=0.00000;faf95_sas=0.00000;faf99=0.00000;faf99_afr=0.00000;faf99_amr=0.00000;faf99_eas=0.00000;faf99_mid=0.00000;faf99_nfe=0.00000;faf99_non_ukb=0.00000;faf99_non_ukb_afr=0.00000;faf99_non_ukb_amr=0.00000;faf99_non_ukb_eas=0.00000;faf99_non_ukb_mid=0.00000;faf99_non_ukb_nfe=0.00000;faf99_non_ukb_sas=0.00000;faf99_sas=0.00000;age_hist_het_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_het_n_smaller=0;age_hist_het_n_larger=0;age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_hom_n_smaller=0;age_hist_hom_n_larger=0;FS=0.00000;MQ=57.9124;MQRankSum=0.00000;QUALapprox=221;QD=20.0909;ReadPosRankSum=0.00000;SOR=0.223144;VarDP=11;AS_FS=0.00000;AS_MQ=60.0000;AS_MQRankSum=0.00000;AS_pab_max=1.00000;AS_QUALapprox=138;AS_QD=19.7143;AS_ReadPosRankSum=0.00000;AS_SB_TABLE=0,1|3,3;AS_SOR=0.223144;AS_VarDP=7;inbreeding_coeff=0.799998;AS_culprit=AS_MQ;AS_VQSLOD=6.28100;negative_train_site;allele_type=snv;n_alt_alleles=2;variant_type=multi-snv;outside_ukb_capture_region;outside_broad_capture_region;gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;gq_hist_all_bin_freq=0|0|0|0|81|0|79|0|7|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_n_larger=0;dp_hist_all_bin_freq=0|0|160|7|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_all_n_larger=0;ab_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;cadd_raw_score=0.473593;cadd_phred=6.22600;spliceai_ds_max=0.00000;pangolin_largest_ds=0.00000;phylop=-0.280000;VRS_Allele_IDs=ga4gh:VA.Atzz2sF20yoKJ1nmL2n_X7Jjmx89Ln5i,ga4gh:VA.xeYLnnwnPXgVc24xU9lxWxWo6MYNcV2A;VRS_Starts=108215365,108215365;VRS_Ends=108215366,108215366;VRS_States=G,T;vep=T|upstream_gene_variant|MODIFIER|ABHD13|ENSG00000139826|Transcript|ENST00000375898|protein_coding||||||||||1|3026|1||SNV|HGNC|HGNC:20293|YES|NM_032859.3||1|P1|CCDS32007.1|ENSP00000365063||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000405925|protein_coding||1/1|ENST00000405925.2:c.-29+2796C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||1|P1|CCDS9508.1|ENSP00000385955||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000442234|protein_coding||1/2|ENST00000442234.6:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|YES|NM_206937.2||1|P1|CCDS9508.1|ENSP00000402030||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000611712|protein_coding||1/2|ENST00000611712.4:c.-299+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||4|P1|CCDS9508.1|ENSP00000484288||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000614526|protein_coding||1/2|ENST00000614526.2:c.-223+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||2||CCDS81779.1|ENSP00000480814||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000685338|protein_coding||1/2|ENST00000685338.1:c.-275+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000510567||Ensembl|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686095|protein_coding||||||||||1|180|-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509942||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686204|protein_coding||1/2|ENST00000686204.1:c.-186+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509685||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686913|protein_coding|1/3||ENST00000686913.1:c.-341C>A||2|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509299||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686926|protein_coding||1/1|ENST00000686926.1:c.-29+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509122||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000687164|protein_coding|1/3||ENST00000687164.1:c.-144C>A||44|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000508512||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000687822|protein_coding||1/2|ENST00000687822.1:c.-210+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509344||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688396|protein_coding||2/3|ENST00000688396.1:c.-210+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509564||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688455|protein_coding||2/3|ENST00000688455.1:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509304||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688529|protein_coding|1/3||ENST00000688529.1:c.-252C>A||36|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509906||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688595|protein_coding||2/3|ENST00000688595.1:c.-186+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509502||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000689762|protein_coding||1/2|ENST00000689762.1:c.-331+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||||CCDS81779.1|ENSP00000508867||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000690127|protein_coding|1/3||ENST00000690127.1:c.-373C>A||22|||||1||-1||SNV|HGNC|HGNC:6601||||||CCDS81779.1|ENSP00000509468||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000692222|protein_coding||1/2|ENST00000692222.1:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509226||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000693040|protein_coding||1/2|ENST00000693040.1:c.-78+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000510014||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001098268.2|protein_coding||1/1|NM_001098268.2:c.-29+2796C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001091738.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001330595.2|protein_coding||1/2|NM_001330595.2:c.-223+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001317524.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352598.2|protein_coding||2/3|NM_001352598.2:c.-186+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339527.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352599.2|protein_coding||2/3|NM_001352599.2:c.-210+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339528.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352600.2|protein_coding||2/3|NM_001352600.2:c.-102+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339529.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352601.2|protein_coding|1/3||NM_001352601.2:c.-252C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339530.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352602.2|protein_coding|1/3||NM_001352602.2:c.-144C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339531.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|3981|Transcript|NM_001352603.1|protein_coding||||||||||1|61|-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339532.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352604.2|protein_coding|1/3||NM_001352604.2:c.-136C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339533.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001379095.1|protein_coding||1/2|NM_001379095.1:c.-299+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001366024.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|3981|Transcript|NM_002312.3|protein_coding||||||||||1|584|-1||SNV|EntrezGene|HGNC:6601|||||||NP_002303.2||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|ABHD13|84945|Transcript|NM_032859.3|protein_coding||||||||||1|3026|1||SNV|EntrezGene|HGNC:20293|YES|ENST00000375898.4|||||NP_116248.2||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_206937.2|protein_coding||1/2|NM_206937.2:c.-102+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|YES|ENST00000442234.6|||||NP_996820.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|ABHD13|84945|Transcript|XM_011521128.3|protein_coding||||||||||1|3380|1||SNV|EntrezGene|HGNC:20293|||||||XP_011519430.1||RefSeq|||||||||||||,T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000273230|promoter||||||||||1||||SNV||||||||||||||||||||||||,T|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00000634950|||||||||||1||1||SNV||||||||||||||||ENSPFM0617|3|Y|-0.094|ZBTB7B&ZBTB7A&ZBTB7C||||\n\n\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:23)\n\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:23)\n\tat is.hail.utils.package$.fatal(package.scala:96)\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2106)\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7$adapted(LoadVCF.scala:2076)\n\tat scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:515)\n\tat scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor_region3_18(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m64split_Block(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$10(BackendUtils.scala:90)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:166)\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$9(BackendUtils.scala:89)\n\tat is.hail.backend.spark.SparkBackend$$anon$5.compute(SparkBackend.scala:444)\n\tat org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)\n\tat org.apache.spark.rdd.RDD.iterator(RDD.scala:331)\n\tat org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)\n\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)\n\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\n\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)\n\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)\n\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\n\nis.hail.utils.HailException: Invalid locus 'chr13:108215366' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\n\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:19)\n\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:19)\n\tat is.hail.utils.package$.fatal(package.scala:96)\n\tat is.hail.variant.ReferenceGenome.checkLocus(ReferenceGenome.scala:237)\n\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3(LoadVCF.scala:521)\n\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3$adapted(LoadVCF.scala:521)\n\tat scala.Option.foreach(Option.scala:407)\n\tat is.hail.io.vcf.VCFLine.parseAddVariant(LoadVCF.scala:521)\n\tat is.hail.io.vcf.LoadVCF$.parseLine(LoadVCF.scala:1483)\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2092)\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7$adapted(LoadVCF.scala:2076)\n\tat scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:515)\n\tat scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor_region3_18(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m64split_Block(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$10(BackendUtils.scala:90)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:166)\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$9(BackendUtils.scala:89)\n\tat is.hail.backend.spark.SparkBackend$$anon$5.compute(SparkBackend.scala:444)\n\tat org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)\n\tat org.apache.spark.rdd.RDD.iterator(RDD.scala:331)\n\tat org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)\n\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)\n\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\n\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)\n\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)\n\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\n\n\n\n\n\nHail version: 0.2.133-4c60fddb171a\nError summary: HailException: Invalid locus 'chr13:108215366' found. Contig 'chr13' is not in the reference genome 'GRCh37'.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFatalError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[8], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInitial dataset dimensions:\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m----> 2\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m- Number of variants: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mmt\u001b[38;5;241m.\u001b[39mcount_rows()\u001b[38;5;132;01m:\u001b[39;00m\u001b[38;5;124m,\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m      3\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m- Number of samples: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mmt\u001b[38;5;241m.\u001b[39mcount_cols()\u001b[38;5;132;01m:\u001b[39;00m\u001b[38;5;124m,\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[1;32mc:\\Users\\jmcdonald\\AppData\\Local\\anaconda3\\envs\\genome-env\\lib\\site-packages\\decorator.py:232\u001b[0m, in \u001b[0;36mdecorate.<locals>.fun\u001b[1;34m(*args, **kw)\u001b[0m\n\u001b[0;32m    230\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m kwsyntax:\n\u001b[0;32m    231\u001b[0m     args, kw \u001b[38;5;241m=\u001b[39m fix(args, kw, sig)\n\u001b[1;32m--> 232\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m caller(func, \u001b[38;5;241m*\u001b[39m(extras \u001b[38;5;241m+\u001b[39m args), \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkw)\n",
      "File \u001b[1;32mc:\\Users\\jmcdonald\\AppData\\Local\\anaconda3\\envs\\genome-env\\lib\\site-packages\\hail\\typecheck\\check.py:585\u001b[0m, in \u001b[0;36m_make_dec.<locals>.wrapper\u001b[1;34m(__original_func, *args, **kwargs)\u001b[0m\n\u001b[0;32m    582\u001b[0m \u001b[38;5;129m@decorator\u001b[39m\n\u001b[0;32m    583\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21mwrapper\u001b[39m(__original_func: Callable[\u001b[38;5;241m.\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;241m.\u001b[39m, T], \u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m T:\n\u001b[0;32m    584\u001b[0m     args_, kwargs_ \u001b[38;5;241m=\u001b[39m check_all(__original_func, args, kwargs, checkers, is_method\u001b[38;5;241m=\u001b[39mis_method)\n\u001b[1;32m--> 585\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m __original_func(\u001b[38;5;241m*\u001b[39margs_, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs_)\n",
      "File \u001b[1;32mc:\\Users\\jmcdonald\\AppData\\Local\\anaconda3\\envs\\genome-env\\lib\\site-packages\\hail\\matrixtable.py:2646\u001b[0m, in \u001b[0;36mMatrixTable.count_rows\u001b[1;34m(self, _localize)\u001b[0m\n\u001b[0;32m   2644\u001b[0m count_ir \u001b[38;5;241m=\u001b[39m ir\u001b[38;5;241m.\u001b[39mTableCount(ir\u001b[38;5;241m.\u001b[39mMatrixRowsTable(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_mir))\n\u001b[0;32m   2645\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m _localize:\n\u001b[1;32m-> 2646\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mEnv\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbackend\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexecute\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcount_ir\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   2647\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m   2648\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m construct_expr(ir\u001b[38;5;241m.\u001b[39mLiftMeOut(count_ir), hl\u001b[38;5;241m.\u001b[39mtint64)\n",
      "File \u001b[1;32mc:\\Users\\jmcdonald\\AppData\\Local\\anaconda3\\envs\\genome-env\\lib\\site-packages\\hail\\backend\\spark_backend.py:217\u001b[0m, in \u001b[0;36mSparkBackend.execute\u001b[1;34m(self, ir, timed)\u001b[0m\n\u001b[0;32m    214\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m fatal:\n\u001b[0;32m    215\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m err \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mfatal\u001b[39;00m\n\u001b[1;32m--> 217\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m err\n",
      "File \u001b[1;32mc:\\Users\\jmcdonald\\AppData\\Local\\anaconda3\\envs\\genome-env\\lib\\site-packages\\hail\\backend\\spark_backend.py:209\u001b[0m, in \u001b[0;36mSparkBackend.execute\u001b[1;34m(self, ir, timed)\u001b[0m\n\u001b[0;32m    207\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21mexecute\u001b[39m(\u001b[38;5;28mself\u001b[39m, ir: BaseIR, timed: \u001b[38;5;28mbool\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Any:\n\u001b[0;32m    208\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 209\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexecute\u001b[49m\u001b[43m(\u001b[49m\u001b[43mir\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimed\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    210\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[0;32m    211\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_copy_log_on_error:\n",
      "File \u001b[1;32mc:\\Users\\jmcdonald\\AppData\\Local\\anaconda3\\envs\\genome-env\\lib\\site-packages\\hail\\backend\\backend.py:181\u001b[0m, in \u001b[0;36mBackend.execute\u001b[1;34m(self, ir, timed)\u001b[0m\n\u001b[0;32m    179\u001b[0m     result, timings \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_rpc(ActionTag\u001b[38;5;241m.\u001b[39mEXECUTE, payload)\n\u001b[0;32m    180\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m FatalError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[1;32m--> 181\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m e\u001b[38;5;241m.\u001b[39mmaybe_user_error(ir) \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m    182\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m ir\u001b[38;5;241m.\u001b[39mtyp \u001b[38;5;241m==\u001b[39m tvoid:\n\u001b[0;32m    183\u001b[0m     value \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\jmcdonald\\AppData\\Local\\anaconda3\\envs\\genome-env\\lib\\site-packages\\hail\\backend\\backend.py:179\u001b[0m, in \u001b[0;36mBackend.execute\u001b[1;34m(self, ir, timed)\u001b[0m\n\u001b[0;32m    177\u001b[0m payload \u001b[38;5;241m=\u001b[39m ExecutePayload(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_render_ir(ir), \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m{\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mname\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m:\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mStreamBufferSpec\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m}\u001b[39m\u001b[38;5;124m'\u001b[39m, timed)\n\u001b[0;32m    178\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 179\u001b[0m     result, timings \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_rpc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mActionTag\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mEXECUTE\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpayload\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    180\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m FatalError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    181\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m e\u001b[38;5;241m.\u001b[39mmaybe_user_error(ir) \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\jmcdonald\\AppData\\Local\\anaconda3\\envs\\genome-env\\lib\\site-packages\\hail\\backend\\py4j_backend.py:221\u001b[0m, in \u001b[0;36mPy4JBackend._rpc\u001b[1;34m(self, action, payload)\u001b[0m\n\u001b[0;32m    219\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m resp\u001b[38;5;241m.\u001b[39mstatus_code \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m400\u001b[39m:\n\u001b[0;32m    220\u001b[0m     error_json \u001b[38;5;241m=\u001b[39m orjson\u001b[38;5;241m.\u001b[39mloads(resp\u001b[38;5;241m.\u001b[39mcontent)\n\u001b[1;32m--> 221\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m fatal_error_from_java_error_triplet(\n\u001b[0;32m    222\u001b[0m         error_json[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mshort\u001b[39m\u001b[38;5;124m'\u001b[39m], error_json[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mexpanded\u001b[39m\u001b[38;5;124m'\u001b[39m], error_json[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124merror_id\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m    223\u001b[0m     )\n\u001b[0;32m    224\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\u001b[38;5;241m.\u001b[39mcontent, resp\u001b[38;5;241m.\u001b[39mheaders\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mX-Hail-Timings\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[1;31mFatalError\u001b[0m: HailException: Invalid locus 'chr13:108215366' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\n\nJava stack trace:\norg.apache.spark.SparkException: Job aborted due to stage failure: Task 22 in stage 0.0 failed 1 times, most recent failure: Lost task 22.0 in stage 0.0 (TID 22) (LT2058-20241028.SHTC.com executor driver): is.hail.utils.HailException: file:/C:/Users/jmcdonald/Desktop/Projects/The Genome Project/gnomad.exomes.v4.1.sites.chr13.vcf.bgz:offset 196318482477539: error while parsing line\nchr13\t108215366\t.\tG\tT\t.\tAC0\tAC=0;AN=334;AF=0.00000;AC_XX=0;AF_XX=0.00000;AN_XX=88;nhomalt_XX=0;AC_XY=0;AF_XY=0.00000;AN_XY=246;nhomalt_XY=0;nhomalt=0;AC_afr_XX=0;AF_afr_XX=0.00000;AN_afr_XX=4;nhomalt_afr_XX=0;AC_afr_XY=0;AF_afr_XY=0.00000;AN_afr_XY=6;nhomalt_afr_XY=0;AC_afr=0;AF_afr=0.00000;AN_afr=10;nhomalt_afr=0;AC_amr_XX=0;AN_amr_XX=0;nhomalt_amr_XX=0;AC_amr_XY=0;AF_amr_XY=0.00000;AN_amr_XY=2;nhomalt_amr_XY=0;AC_amr=0;AF_amr=0.00000;AN_amr=2;nhomalt_amr=0;AC_asj_XX=0;AN_asj_XX=0;nhomalt_asj_XX=0;AC_asj_XY=0;AF_asj_XY=0.00000;AN_asj_XY=2;nhomalt_asj_XY=0;AC_asj=0;AF_asj=0.00000;AN_asj=2;nhomalt_asj=0;AC_eas_XX=0;AN_eas_XX=0;nhomalt_eas_XX=0;AC_eas_XY=0;AF_eas_XY=0.00000;AN_eas_XY=14;nhomalt_eas_XY=0;AC_eas=0;AF_eas=0.00000;AN_eas=14;nhomalt_eas=0;AC_fin_XX=0;AN_fin_XX=0;nhomalt_fin_XX=0;AC_fin_XY=0;AN_fin_XY=0;nhomalt_fin_XY=0;AC_fin=0;AN_fin=0;nhomalt_fin=0;AC_mid_XX=0;AN_mid_XX=0;nhomalt_mid_XX=0;AC_mid_XY=0;AN_mid_XY=0;nhomalt_mid_XY=0;AC_mid=0;AN_mid=0;nhomalt_mid=0;AC_nfe_XX=0;AF_nfe_XX=0.00000;AN_nfe_XX=74;nhomalt_nfe_XX=0;AC_nfe_XY=0;AF_nfe_XY=0.00000;AN_nfe_XY=204;nhomalt_nfe_XY=0;AC_nfe=0;AF_nfe=0.00000;AN_nfe=278;nhomalt_nfe=0;AC_non_ukb_XX=0;AF_non_ukb_XX=0.00000;AN_non_ukb_XX=88;nhomalt_non_ukb_XX=0;AC_non_ukb_XY=0;AF_non_ukb_XY=0.00000;AN_non_ukb_XY=246;nhomalt_non_ukb_XY=0;AC_non_ukb=0;AF_non_ukb=0.00000;AN_non_ukb=334;nhomalt_non_ukb=0;AC_non_ukb_afr_XX=0;AF_non_ukb_afr_XX=0.00000;AN_non_ukb_afr_XX=4;nhomalt_non_ukb_afr_XX=0;AC_non_ukb_afr_XY=0;AF_non_ukb_afr_XY=0.00000;AN_non_ukb_afr_XY=6;nhomalt_non_ukb_afr_XY=0;AC_non_ukb_afr=0;AF_non_ukb_afr=0.00000;AN_non_ukb_afr=10;nhomalt_non_ukb_afr=0;AC_non_ukb_amr_XX=0;AN_non_ukb_amr_XX=0;nhomalt_non_ukb_amr_XX=0;AC_non_ukb_amr_XY=0;AF_non_ukb_amr_XY=0.00000;AN_non_ukb_amr_XY=2;nhomalt_non_ukb_amr_XY=0;AC_non_ukb_amr=0;AF_non_ukb_amr=0.00000;AN_non_ukb_amr=2;nhomalt_non_ukb_amr=0;AC_non_ukb_asj_XX=0;AN_non_ukb_asj_XX=0;nhomalt_non_ukb_asj_XX=0;AC_non_ukb_asj_XY=0;AF_non_ukb_asj_XY=0.00000;AN_non_ukb_asj_XY=2;nhomalt_non_ukb_asj_XY=0;AC_non_ukb_asj=0;AF_non_ukb_asj=0.00000;AN_non_ukb_asj=2;nhomalt_non_ukb_asj=0;AC_non_ukb_eas_XX=0;AN_non_ukb_eas_XX=0;nhomalt_non_ukb_eas_XX=0;AC_non_ukb_eas_XY=0;AF_non_ukb_eas_XY=0.00000;AN_non_ukb_eas_XY=14;nhomalt_non_ukb_eas_XY=0;AC_non_ukb_eas=0;AF_non_ukb_eas=0.00000;AN_non_ukb_eas=14;nhomalt_non_ukb_eas=0;AC_non_ukb_fin_XX=0;AN_non_ukb_fin_XX=0;nhomalt_non_ukb_fin_XX=0;AC_non_ukb_fin_XY=0;AN_non_ukb_fin_XY=0;nhomalt_non_ukb_fin_XY=0;AC_non_ukb_fin=0;AN_non_ukb_fin=0;nhomalt_non_ukb_fin=0;AC_non_ukb_mid_XX=0;AN_non_ukb_mid_XX=0;nhomalt_non_ukb_mid_XX=0;AC_non_ukb_mid_XY=0;AN_non_ukb_mid_XY=0;nhomalt_non_ukb_mid_XY=0;AC_non_ukb_mid=0;AN_non_ukb_mid=0;nhomalt_non_ukb_mid=0;AC_non_ukb_nfe_XX=0;AF_non_ukb_nfe_XX=0.00000;AN_non_ukb_nfe_XX=74;nhomalt_non_ukb_nfe_XX=0;AC_non_ukb_nfe_XY=0;AF_non_ukb_nfe_XY=0.00000;AN_non_ukb_nfe_XY=204;nhomalt_non_ukb_nfe_XY=0;AC_non_ukb_nfe=0;AF_non_ukb_nfe=0.00000;AN_non_ukb_nfe=278;nhomalt_non_ukb_nfe=0;AC_non_ukb_raw=5;AF_non_ukb_raw=7.95186e-06;AN_non_ukb_raw=628784;nhomalt_non_ukb_raw=2;AC_non_ukb_remaining_XX=0;AF_non_ukb_remaining_XX=0.00000;AN_non_ukb_remaining_XX=8;nhomalt_non_ukb_remaining_XX=0;AC_non_ukb_remaining_XY=0;AF_non_ukb_remaining_XY=0.00000;AN_non_ukb_remaining_XY=12;nhomalt_non_ukb_remaining_XY=0;AC_non_ukb_remaining=0;AF_non_ukb_remaining=0.00000;AN_non_ukb_remaining=20;nhomalt_non_ukb_remaining=0;AC_non_ukb_sas_XX=0;AF_non_ukb_sas_XX=0.00000;AN_non_ukb_sas_XX=2;nhomalt_non_ukb_sas_XX=0;AC_non_ukb_sas_XY=0;AF_non_ukb_sas_XY=0.00000;AN_non_ukb_sas_XY=6;nhomalt_non_ukb_sas_XY=0;AC_non_ukb_sas=0;AF_non_ukb_sas=0.00000;AN_non_ukb_sas=8;nhomalt_non_ukb_sas=0;AC_raw=5;AF_raw=7.94690e-06;AN_raw=629176;nhomalt_raw=2;AC_remaining_XX=0;AF_remaining_XX=0.00000;AN_remaining_XX=8;nhomalt_remaining_XX=0;AC_remaining_XY=0;AF_remaining_XY=0.00000;AN_remaining_XY=12;nhomalt_remaining_XY=0;AC_remaining=0;AF_remaining=0.00000;AN_remaining=20;nhomalt_remaining=0;AC_sas_XX=0;AF_sas_XX=0.00000;AN_sas_XX=2;nhomalt_sas_XX=0;AC_sas_XY=0;AF_sas_XY=0.00000;AN_sas_XY=6;nhomalt_sas_XY=0;AC_sas=0;AF_sas=0.00000;AN_sas=8;nhomalt_sas=0;faf95=0.00000;faf95_afr=0.00000;faf95_amr=0.00000;faf95_eas=0.00000;faf95_mid=0.00000;faf95_nfe=0.00000;faf95_non_ukb=0.00000;faf95_non_ukb_afr=0.00000;faf95_non_ukb_amr=0.00000;faf95_non_ukb_eas=0.00000;faf95_non_ukb_mid=0.00000;faf95_non_ukb_nfe=0.00000;faf95_non_ukb_sas=0.00000;faf95_sas=0.00000;faf99=0.00000;faf99_afr=0.00000;faf99_amr=0.00000;faf99_eas=0.00000;faf99_mid=0.00000;faf99_nfe=0.00000;faf99_non_ukb=0.00000;faf99_non_ukb_afr=0.00000;faf99_non_ukb_amr=0.00000;faf99_non_ukb_eas=0.00000;faf99_non_ukb_mid=0.00000;faf99_non_ukb_nfe=0.00000;faf99_non_ukb_sas=0.00000;faf99_sas=0.00000;age_hist_het_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_het_n_smaller=0;age_hist_het_n_larger=0;age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_hom_n_smaller=0;age_hist_hom_n_larger=0;FS=0.00000;MQ=57.9124;MQRankSum=0.00000;QUALapprox=221;QD=20.0909;ReadPosRankSum=0.00000;SOR=0.223144;VarDP=11;AS_FS=0.00000;AS_MQ=60.0000;AS_MQRankSum=0.00000;AS_pab_max=1.00000;AS_QUALapprox=138;AS_QD=19.7143;AS_ReadPosRankSum=0.00000;AS_SB_TABLE=0,1|3,3;AS_SOR=0.223144;AS_VarDP=7;inbreeding_coeff=0.799998;AS_culprit=AS_MQ;AS_VQSLOD=6.28100;negative_train_site;allele_type=snv;n_alt_alleles=2;variant_type=multi-snv;outside_ukb_capture_region;outside_broad_capture_region;gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;gq_hist_all_bin_freq=0|0|0|0|81|0|79|0|7|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_n_larger=0;dp_hist_all_bin_freq=0|0|160|7|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_all_n_larger=0;ab_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;cadd_raw_score=0.473593;cadd_phred=6.22600;spliceai_ds_max=0.00000;pangolin_largest_ds=0.00000;phylop=-0.280000;VRS_Allele_IDs=ga4gh:VA.Atzz2sF20yoKJ1nmL2n_X7Jjmx89Ln5i,ga4gh:VA.xeYLnnwnPXgVc24xU9lxWxWo6MYNcV2A;VRS_Starts=108215365,108215365;VRS_Ends=108215366,108215366;VRS_States=G,T;vep=T|upstream_gene_variant|MODIFIER|ABHD13|ENSG00000139826|Transcript|ENST00000375898|protein_coding||||||||||1|3026|1||SNV|HGNC|HGNC:20293|YES|NM_032859.3||1|P1|CCDS32007.1|ENSP00000365063||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000405925|protein_coding||1/1|ENST00000405925.2:c.-29+2796C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||1|P1|CCDS9508.1|ENSP00000385955||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000442234|protein_coding||1/2|ENST00000442234.6:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|YES|NM_206937.2||1|P1|CCDS9508.1|ENSP00000402030||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000611712|protein_coding||1/2|ENST00000611712.4:c.-299+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||4|P1|CCDS9508.1|ENSP00000484288||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000614526|protein_coding||1/2|ENST00000614526.2:c.-223+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||2||CCDS81779.1|ENSP00000480814||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000685338|protein_coding||1/2|ENST00000685338.1:c.-275+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000510567||Ensembl|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686095|protein_coding||||||||||1|180|-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509942||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686204|protein_coding||1/2|ENST00000686204.1:c.-186+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509685||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686913|protein_coding|1/3||ENST00000686913.1:c.-341C>A||2|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509299||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686926|protein_coding||1/1|ENST00000686926.1:c.-29+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509122||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000687164|protein_coding|1/3||ENST00000687164.1:c.-144C>A||44|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000508512||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000687822|protein_coding||1/2|ENST00000687822.1:c.-210+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509344||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688396|protein_coding||2/3|ENST00000688396.1:c.-210+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509564||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688455|protein_coding||2/3|ENST00000688455.1:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509304||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688529|protein_coding|1/3||ENST00000688529.1:c.-252C>A||36|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509906||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688595|protein_coding||2/3|ENST00000688595.1:c.-186+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509502||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000689762|protein_coding||1/2|ENST00000689762.1:c.-331+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||||CCDS81779.1|ENSP00000508867||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000690127|protein_coding|1/3||ENST00000690127.1:c.-373C>A||22|||||1||-1||SNV|HGNC|HGNC:6601||||||CCDS81779.1|ENSP00000509468||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000692222|protein_coding||1/2|ENST00000692222.1:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509226||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000693040|protein_coding||1/2|ENST00000693040.1:c.-78+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000510014||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001098268.2|protein_coding||1/1|NM_001098268.2:c.-29+2796C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001091738.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001330595.2|protein_coding||1/2|NM_001330595.2:c.-223+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001317524.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352598.2|protein_coding||2/3|NM_001352598.2:c.-186+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339527.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352599.2|protein_coding||2/3|NM_001352599.2:c.-210+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339528.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352600.2|protein_coding||2/3|NM_001352600.2:c.-102+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339529.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352601.2|protein_coding|1/3||NM_001352601.2:c.-252C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339530.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352602.2|protein_coding|1/3||NM_001352602.2:c.-144C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339531.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|3981|Transcript|NM_001352603.1|protein_coding||||||||||1|61|-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339532.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352604.2|protein_coding|1/3||NM_001352604.2:c.-136C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339533.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001379095.1|protein_coding||1/2|NM_001379095.1:c.-299+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001366024.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|3981|Transcript|NM_002312.3|protein_coding||||||||||1|584|-1||SNV|EntrezGene|HGNC:6601|||||||NP_002303.2||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|ABHD13|84945|Transcript|NM_032859.3|protein_coding||||||||||1|3026|1||SNV|EntrezGene|HGNC:20293|YES|ENST00000375898.4|||||NP_116248.2||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_206937.2|protein_coding||1/2|NM_206937.2:c.-102+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|YES|ENST00000442234.6|||||NP_996820.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|ABHD13|84945|Transcript|XM_011521128.3|protein_coding||||||||||1|3380|1||SNV|EntrezGene|HGNC:20293|||||||XP_011519430.1||RefSeq|||||||||||||,T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000273230|promoter||||||||||1||||SNV||||||||||||||||||||||||,T|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00000634950|||||||||||1||1||SNV||||||||||||||||ENSPFM0617|3|Y|-0.094|ZBTB7B&ZBTB7A&ZBTB7C||||\n\r\n\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:23)\r\n\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:23)\r\n\tat is.hail.utils.package$.fatal(package.scala:96)\r\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2106)\r\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7$adapted(LoadVCF.scala:2076)\r\n\tat scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:515)\r\n\tat scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor_region3_18(Unknown Source)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor(Unknown Source)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m64split_Block(Unknown Source)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\r\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\r\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$10(BackendUtils.scala:90)\r\n\tat is.hail.utils.package$.using(package.scala:673)\r\n\tat is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:166)\r\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$9(BackendUtils.scala:89)\r\n\tat is.hail.backend.spark.SparkBackend$$anon$5.compute(SparkBackend.scala:444)\r\n\tat org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)\r\n\tat org.apache.spark.rdd.RDD.iterator(RDD.scala:331)\r\n\tat org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)\r\n\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)\r\n\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\r\n\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)\r\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)\r\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)\r\n\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)\r\n\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\r\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\r\nCaused by: is.hail.utils.HailException: Invalid locus 'chr13:108215366' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\r\n\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:19)\r\n\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:19)\r\n\tat is.hail.utils.package$.fatal(package.scala:96)\r\n\tat is.hail.variant.ReferenceGenome.checkLocus(ReferenceGenome.scala:237)\r\n\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3(LoadVCF.scala:521)\r\n\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3$adapted(LoadVCF.scala:521)\r\n\tat scala.Option.foreach(Option.scala:407)\r\n\tat is.hail.io.vcf.VCFLine.parseAddVariant(LoadVCF.scala:521)\r\n\tat is.hail.io.vcf.LoadVCF$.parseLine(LoadVCF.scala:1483)\r\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2092)\r\n\t... 26 more\r\n\nDriver stacktrace:\n\tat org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)\n\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n\tat org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)\n\tat org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)\n\tat scala.Option.foreach(Option.scala:407)\n\tat org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)\n\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)\n\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)\n\tat org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)\n\tat org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)\n\tat org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)\n\tat org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$parallelizeAndComputeWithIndex$4(SparkBackend.scala:455)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$parallelizeAndComputeWithIndex$4$adapted(SparkBackend.scala:454)\n\tat scala.collection.Iterator.foreach(Iterator.scala:943)\n\tat scala.collection.Iterator.foreach$(Iterator.scala:943)\n\tat scala.collection.AbstractIterator.foreach(Iterator.scala:1431)\n\tat is.hail.backend.spark.SparkBackend.parallelizeAndComputeWithIndex(SparkBackend.scala:454)\n\tat is.hail.backend.BackendUtils.collectDArray(BackendUtils.scala:85)\n\tat __C41Compiled.apply(Emit.scala)\n\tat is.hail.expr.ir.LoweredTableReader$.$anonfun$makeCoercer$15(TableIR.scala:402)\n\tat is.hail.backend.ExecuteContext.$anonfun$scopedExecution$1(ExecuteContext.scala:144)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.backend.ExecuteContext.scopedExecution(ExecuteContext.scala:144)\n\tat is.hail.expr.ir.LoweredTableReader$.makeCoercer(TableIR.scala:401)\n\tat is.hail.expr.ir.GenericTableValue.getLTVCoercer(GenericTableValue.scala:187)\n\tat is.hail.expr.ir.GenericTableValue.toTableStage(GenericTableValue.scala:220)\n\tat is.hail.io.vcf.MatrixVCFReader.lower(LoadVCF.scala:2140)\n\tat is.hail.expr.ir.TableReader.lower(TableIR.scala:610)\n\tat is.hail.expr.ir.lowering.LowerTableIR$.applyTable(LowerTableIR.scala:1062)\n\tat is.hail.expr.ir.lowering.LowerTableIR$.lower$1(LowerTableIR.scala:728)\n\tat is.hail.expr.ir.lowering.LowerTableIR$.apply(LowerTableIR.scala:732)\n\tat is.hail.expr.ir.lowering.LowerToCDA$.lower(LowerToCDA.scala:27)\n\tat is.hail.expr.ir.lowering.LowerToCDA$.apply(LowerToCDA.scala:11)\n\tat is.hail.expr.ir.lowering.LowerToDistributedArrayPass.transform(LoweringPass.scala:91)\n\tat is.hail.expr.ir.LowerOrInterpretNonCompilable$.evaluate$1(LowerOrInterpretNonCompilable.scala:27)\n\tat is.hail.expr.ir.LowerOrInterpretNonCompilable$.rewrite$1(LowerOrInterpretNonCompilable.scala:59)\n\tat is.hail.expr.ir.LowerOrInterpretNonCompilable$.apply(LowerOrInterpretNonCompilable.scala:64)\n\tat is.hail.expr.ir.lowering.LowerOrInterpretNonCompilablePass$.transform(LoweringPass.scala:83)\n\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$3(LoweringPass.scala:32)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$1(LoweringPass.scala:32)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.expr.ir.lowering.LoweringPass.apply(LoweringPass.scala:30)\n\tat is.hail.expr.ir.lowering.LoweringPass.apply$(LoweringPass.scala:29)\n\tat is.hail.expr.ir.lowering.LowerOrInterpretNonCompilablePass$.apply(LoweringPass.scala:78)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1(LoweringPipeline.scala:21)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1$adapted(LoweringPipeline.scala:19)\n\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.apply(LoweringPipeline.scala:19)\n\tat is.hail.expr.ir.lowering.EvalRelationalLets$.execute$1(EvalRelationalLets.scala:13)\n\tat is.hail.expr.ir.lowering.EvalRelationalLets$.lower$1(EvalRelationalLets.scala:21)\n\tat is.hail.expr.ir.lowering.EvalRelationalLets$.apply(EvalRelationalLets.scala:35)\n\tat is.hail.expr.ir.lowering.EvalRelationalLetsPass.transform(LoweringPass.scala:168)\n\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$3(LoweringPass.scala:32)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.expr.ir.lowering.LoweringPass.$anonfun$apply$1(LoweringPass.scala:32)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.expr.ir.lowering.LoweringPass.apply(LoweringPass.scala:30)\n\tat is.hail.expr.ir.lowering.LoweringPass.apply$(LoweringPass.scala:29)\n\tat is.hail.expr.ir.lowering.EvalRelationalLetsPass.apply(LoweringPass.scala:162)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1(LoweringPipeline.scala:21)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.$anonfun$apply$1$adapted(LoweringPipeline.scala:19)\n\tat scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)\n\tat scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)\n\tat scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)\n\tat is.hail.expr.ir.lowering.LoweringPipeline.apply(LoweringPipeline.scala:19)\n\tat is.hail.expr.ir.CompileAndEvaluate$._apply(CompileAndEvaluate.scala:45)\n\tat is.hail.backend.spark.SparkBackend._execute(SparkBackend.scala:578)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$execute$4(SparkBackend.scala:614)\n\tat is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:84)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$execute$3(SparkBackend.scala:609)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$execute$3$adapted(SparkBackend.scala:608)\n\tat is.hail.backend.ExecuteContext$.$anonfun$scoped$3(ExecuteContext.scala:78)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.backend.ExecuteContext$.$anonfun$scoped$2(ExecuteContext.scala:78)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.annotations.RegionPool$.scoped(RegionPool.scala:13)\n\tat is.hail.backend.ExecuteContext$.scoped(ExecuteContext.scala:65)\n\tat is.hail.backend.spark.SparkBackend.$anonfun$withExecuteContext$2(SparkBackend.scala:411)\n\tat is.hail.utils.ExecutionTimer$.time(ExecutionTimer.scala:55)\n\tat is.hail.utils.ExecutionTimer$.logTime(ExecutionTimer.scala:62)\n\tat is.hail.backend.spark.SparkBackend.withExecuteContext(SparkBackend.scala:397)\n\tat is.hail.backend.spark.SparkBackend.execute(SparkBackend.scala:608)\n\tat is.hail.backend.BackendHttpHandler.handle(BackendServer.scala:88)\n\tat jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)\n\tat jdk.httpserver/sun.net.httpserver.AuthFilter.doFilter(AuthFilter.java:82)\n\tat jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:101)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Exchange$LinkHandler.handle(ServerImpl.java:871)\n\tat jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Exchange.run(ServerImpl.java:847)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$DefaultExecutor.execute(ServerImpl.java:203)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Dispatcher.handle(ServerImpl.java:566)\n\tat jdk.httpserver/sun.net.httpserver.ServerImpl$Dispatcher.run(ServerImpl.java:531)\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\n\nis.hail.utils.HailException: file:/C:/Users/jmcdonald/Desktop/Projects/The Genome Project/gnomad.exomes.v4.1.sites.chr13.vcf.bgz:offset 196318482477539: error while parsing line\nchr13\t108215366\t.\tG\tT\t.\tAC0\tAC=0;AN=334;AF=0.00000;AC_XX=0;AF_XX=0.00000;AN_XX=88;nhomalt_XX=0;AC_XY=0;AF_XY=0.00000;AN_XY=246;nhomalt_XY=0;nhomalt=0;AC_afr_XX=0;AF_afr_XX=0.00000;AN_afr_XX=4;nhomalt_afr_XX=0;AC_afr_XY=0;AF_afr_XY=0.00000;AN_afr_XY=6;nhomalt_afr_XY=0;AC_afr=0;AF_afr=0.00000;AN_afr=10;nhomalt_afr=0;AC_amr_XX=0;AN_amr_XX=0;nhomalt_amr_XX=0;AC_amr_XY=0;AF_amr_XY=0.00000;AN_amr_XY=2;nhomalt_amr_XY=0;AC_amr=0;AF_amr=0.00000;AN_amr=2;nhomalt_amr=0;AC_asj_XX=0;AN_asj_XX=0;nhomalt_asj_XX=0;AC_asj_XY=0;AF_asj_XY=0.00000;AN_asj_XY=2;nhomalt_asj_XY=0;AC_asj=0;AF_asj=0.00000;AN_asj=2;nhomalt_asj=0;AC_eas_XX=0;AN_eas_XX=0;nhomalt_eas_XX=0;AC_eas_XY=0;AF_eas_XY=0.00000;AN_eas_XY=14;nhomalt_eas_XY=0;AC_eas=0;AF_eas=0.00000;AN_eas=14;nhomalt_eas=0;AC_fin_XX=0;AN_fin_XX=0;nhomalt_fin_XX=0;AC_fin_XY=0;AN_fin_XY=0;nhomalt_fin_XY=0;AC_fin=0;AN_fin=0;nhomalt_fin=0;AC_mid_XX=0;AN_mid_XX=0;nhomalt_mid_XX=0;AC_mid_XY=0;AN_mid_XY=0;nhomalt_mid_XY=0;AC_mid=0;AN_mid=0;nhomalt_mid=0;AC_nfe_XX=0;AF_nfe_XX=0.00000;AN_nfe_XX=74;nhomalt_nfe_XX=0;AC_nfe_XY=0;AF_nfe_XY=0.00000;AN_nfe_XY=204;nhomalt_nfe_XY=0;AC_nfe=0;AF_nfe=0.00000;AN_nfe=278;nhomalt_nfe=0;AC_non_ukb_XX=0;AF_non_ukb_XX=0.00000;AN_non_ukb_XX=88;nhomalt_non_ukb_XX=0;AC_non_ukb_XY=0;AF_non_ukb_XY=0.00000;AN_non_ukb_XY=246;nhomalt_non_ukb_XY=0;AC_non_ukb=0;AF_non_ukb=0.00000;AN_non_ukb=334;nhomalt_non_ukb=0;AC_non_ukb_afr_XX=0;AF_non_ukb_afr_XX=0.00000;AN_non_ukb_afr_XX=4;nhomalt_non_ukb_afr_XX=0;AC_non_ukb_afr_XY=0;AF_non_ukb_afr_XY=0.00000;AN_non_ukb_afr_XY=6;nhomalt_non_ukb_afr_XY=0;AC_non_ukb_afr=0;AF_non_ukb_afr=0.00000;AN_non_ukb_afr=10;nhomalt_non_ukb_afr=0;AC_non_ukb_amr_XX=0;AN_non_ukb_amr_XX=0;nhomalt_non_ukb_amr_XX=0;AC_non_ukb_amr_XY=0;AF_non_ukb_amr_XY=0.00000;AN_non_ukb_amr_XY=2;nhomalt_non_ukb_amr_XY=0;AC_non_ukb_amr=0;AF_non_ukb_amr=0.00000;AN_non_ukb_amr=2;nhomalt_non_ukb_amr=0;AC_non_ukb_asj_XX=0;AN_non_ukb_asj_XX=0;nhomalt_non_ukb_asj_XX=0;AC_non_ukb_asj_XY=0;AF_non_ukb_asj_XY=0.00000;AN_non_ukb_asj_XY=2;nhomalt_non_ukb_asj_XY=0;AC_non_ukb_asj=0;AF_non_ukb_asj=0.00000;AN_non_ukb_asj=2;nhomalt_non_ukb_asj=0;AC_non_ukb_eas_XX=0;AN_non_ukb_eas_XX=0;nhomalt_non_ukb_eas_XX=0;AC_non_ukb_eas_XY=0;AF_non_ukb_eas_XY=0.00000;AN_non_ukb_eas_XY=14;nhomalt_non_ukb_eas_XY=0;AC_non_ukb_eas=0;AF_non_ukb_eas=0.00000;AN_non_ukb_eas=14;nhomalt_non_ukb_eas=0;AC_non_ukb_fin_XX=0;AN_non_ukb_fin_XX=0;nhomalt_non_ukb_fin_XX=0;AC_non_ukb_fin_XY=0;AN_non_ukb_fin_XY=0;nhomalt_non_ukb_fin_XY=0;AC_non_ukb_fin=0;AN_non_ukb_fin=0;nhomalt_non_ukb_fin=0;AC_non_ukb_mid_XX=0;AN_non_ukb_mid_XX=0;nhomalt_non_ukb_mid_XX=0;AC_non_ukb_mid_XY=0;AN_non_ukb_mid_XY=0;nhomalt_non_ukb_mid_XY=0;AC_non_ukb_mid=0;AN_non_ukb_mid=0;nhomalt_non_ukb_mid=0;AC_non_ukb_nfe_XX=0;AF_non_ukb_nfe_XX=0.00000;AN_non_ukb_nfe_XX=74;nhomalt_non_ukb_nfe_XX=0;AC_non_ukb_nfe_XY=0;AF_non_ukb_nfe_XY=0.00000;AN_non_ukb_nfe_XY=204;nhomalt_non_ukb_nfe_XY=0;AC_non_ukb_nfe=0;AF_non_ukb_nfe=0.00000;AN_non_ukb_nfe=278;nhomalt_non_ukb_nfe=0;AC_non_ukb_raw=5;AF_non_ukb_raw=7.95186e-06;AN_non_ukb_raw=628784;nhomalt_non_ukb_raw=2;AC_non_ukb_remaining_XX=0;AF_non_ukb_remaining_XX=0.00000;AN_non_ukb_remaining_XX=8;nhomalt_non_ukb_remaining_XX=0;AC_non_ukb_remaining_XY=0;AF_non_ukb_remaining_XY=0.00000;AN_non_ukb_remaining_XY=12;nhomalt_non_ukb_remaining_XY=0;AC_non_ukb_remaining=0;AF_non_ukb_remaining=0.00000;AN_non_ukb_remaining=20;nhomalt_non_ukb_remaining=0;AC_non_ukb_sas_XX=0;AF_non_ukb_sas_XX=0.00000;AN_non_ukb_sas_XX=2;nhomalt_non_ukb_sas_XX=0;AC_non_ukb_sas_XY=0;AF_non_ukb_sas_XY=0.00000;AN_non_ukb_sas_XY=6;nhomalt_non_ukb_sas_XY=0;AC_non_ukb_sas=0;AF_non_ukb_sas=0.00000;AN_non_ukb_sas=8;nhomalt_non_ukb_sas=0;AC_raw=5;AF_raw=7.94690e-06;AN_raw=629176;nhomalt_raw=2;AC_remaining_XX=0;AF_remaining_XX=0.00000;AN_remaining_XX=8;nhomalt_remaining_XX=0;AC_remaining_XY=0;AF_remaining_XY=0.00000;AN_remaining_XY=12;nhomalt_remaining_XY=0;AC_remaining=0;AF_remaining=0.00000;AN_remaining=20;nhomalt_remaining=0;AC_sas_XX=0;AF_sas_XX=0.00000;AN_sas_XX=2;nhomalt_sas_XX=0;AC_sas_XY=0;AF_sas_XY=0.00000;AN_sas_XY=6;nhomalt_sas_XY=0;AC_sas=0;AF_sas=0.00000;AN_sas=8;nhomalt_sas=0;faf95=0.00000;faf95_afr=0.00000;faf95_amr=0.00000;faf95_eas=0.00000;faf95_mid=0.00000;faf95_nfe=0.00000;faf95_non_ukb=0.00000;faf95_non_ukb_afr=0.00000;faf95_non_ukb_amr=0.00000;faf95_non_ukb_eas=0.00000;faf95_non_ukb_mid=0.00000;faf95_non_ukb_nfe=0.00000;faf95_non_ukb_sas=0.00000;faf95_sas=0.00000;faf99=0.00000;faf99_afr=0.00000;faf99_amr=0.00000;faf99_eas=0.00000;faf99_mid=0.00000;faf99_nfe=0.00000;faf99_non_ukb=0.00000;faf99_non_ukb_afr=0.00000;faf99_non_ukb_amr=0.00000;faf99_non_ukb_eas=0.00000;faf99_non_ukb_mid=0.00000;faf99_non_ukb_nfe=0.00000;faf99_non_ukb_sas=0.00000;faf99_sas=0.00000;age_hist_het_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_het_n_smaller=0;age_hist_het_n_larger=0;age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_hom_n_smaller=0;age_hist_hom_n_larger=0;FS=0.00000;MQ=57.9124;MQRankSum=0.00000;QUALapprox=221;QD=20.0909;ReadPosRankSum=0.00000;SOR=0.223144;VarDP=11;AS_FS=0.00000;AS_MQ=60.0000;AS_MQRankSum=0.00000;AS_pab_max=1.00000;AS_QUALapprox=138;AS_QD=19.7143;AS_ReadPosRankSum=0.00000;AS_SB_TABLE=0,1|3,3;AS_SOR=0.223144;AS_VarDP=7;inbreeding_coeff=0.799998;AS_culprit=AS_MQ;AS_VQSLOD=6.28100;negative_train_site;allele_type=snv;n_alt_alleles=2;variant_type=multi-snv;outside_ukb_capture_region;outside_broad_capture_region;gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;gq_hist_all_bin_freq=0|0|0|0|81|0|79|0|7|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_n_larger=0;dp_hist_all_bin_freq=0|0|160|7|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_all_n_larger=0;ab_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;cadd_raw_score=0.473593;cadd_phred=6.22600;spliceai_ds_max=0.00000;pangolin_largest_ds=0.00000;phylop=-0.280000;VRS_Allele_IDs=ga4gh:VA.Atzz2sF20yoKJ1nmL2n_X7Jjmx89Ln5i,ga4gh:VA.xeYLnnwnPXgVc24xU9lxWxWo6MYNcV2A;VRS_Starts=108215365,108215365;VRS_Ends=108215366,108215366;VRS_States=G,T;vep=T|upstream_gene_variant|MODIFIER|ABHD13|ENSG00000139826|Transcript|ENST00000375898|protein_coding||||||||||1|3026|1||SNV|HGNC|HGNC:20293|YES|NM_032859.3||1|P1|CCDS32007.1|ENSP00000365063||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000405925|protein_coding||1/1|ENST00000405925.2:c.-29+2796C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||1|P1|CCDS9508.1|ENSP00000385955||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000442234|protein_coding||1/2|ENST00000442234.6:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|YES|NM_206937.2||1|P1|CCDS9508.1|ENSP00000402030||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000611712|protein_coding||1/2|ENST00000611712.4:c.-299+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||4|P1|CCDS9508.1|ENSP00000484288||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000614526|protein_coding||1/2|ENST00000614526.2:c.-223+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||2||CCDS81779.1|ENSP00000480814||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000685338|protein_coding||1/2|ENST00000685338.1:c.-275+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000510567||Ensembl|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686095|protein_coding||||||||||1|180|-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509942||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686204|protein_coding||1/2|ENST00000686204.1:c.-186+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509685||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686913|protein_coding|1/3||ENST00000686913.1:c.-341C>A||2|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509299||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000686926|protein_coding||1/1|ENST00000686926.1:c.-29+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509122||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000687164|protein_coding|1/3||ENST00000687164.1:c.-144C>A||44|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000508512||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000687822|protein_coding||1/2|ENST00000687822.1:c.-210+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509344||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688396|protein_coding||2/3|ENST00000688396.1:c.-210+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509564||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688455|protein_coding||2/3|ENST00000688455.1:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509304||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688529|protein_coding|1/3||ENST00000688529.1:c.-252C>A||36|||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509906||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000688595|protein_coding||2/3|ENST00000688595.1:c.-186+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509502||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000689762|protein_coding||1/2|ENST00000689762.1:c.-331+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601||||||CCDS81779.1|ENSP00000508867||Ensembl|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000690127|protein_coding|1/3||ENST00000690127.1:c.-373C>A||22|||||1||-1||SNV|HGNC|HGNC:6601||||||CCDS81779.1|ENSP00000509468||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000692222|protein_coding||1/2|ENST00000692222.1:c.-102+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000509226||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|ENSG00000174405|Transcript|ENST00000693040|protein_coding||1/2|ENST00000693040.1:c.-78+118C>A|||||||1||-1||SNV|HGNC|HGNC:6601|||||P1|CCDS9508.1|ENSP00000510014||Ensembl|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001098268.2|protein_coding||1/1|NM_001098268.2:c.-29+2796C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001091738.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001330595.2|protein_coding||1/2|NM_001330595.2:c.-223+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001317524.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352598.2|protein_coding||2/3|NM_001352598.2:c.-186+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339527.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352599.2|protein_coding||2/3|NM_001352599.2:c.-210+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339528.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001352600.2|protein_coding||2/3|NM_001352600.2:c.-102+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339529.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352601.2|protein_coding|1/3||NM_001352601.2:c.-252C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339530.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352602.2|protein_coding|1/3||NM_001352602.2:c.-144C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339531.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|3981|Transcript|NM_001352603.1|protein_coding||||||||||1|61|-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339532.1||RefSeq|||||||||||||,T|5_prime_UTR_variant|MODIFIER|LIG4|3981|Transcript|NM_001352604.2|protein_coding|1/3||NM_001352604.2:c.-136C>A||7|||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001339533.1||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_001379095.1|protein_coding||1/2|NM_001379095.1:c.-299+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|||||||NP_001366024.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|LIG4|3981|Transcript|NM_002312.3|protein_coding||||||||||1|584|-1||SNV|EntrezGene|HGNC:6601|||||||NP_002303.2||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|ABHD13|84945|Transcript|NM_032859.3|protein_coding||||||||||1|3026|1||SNV|EntrezGene|HGNC:20293|YES|ENST00000375898.4|||||NP_116248.2||RefSeq|||||||||||||,T|intron_variant|MODIFIER|LIG4|3981|Transcript|NM_206937.2|protein_coding||1/2|NM_206937.2:c.-102+118C>A|||||||1||-1||SNV|EntrezGene|HGNC:6601|YES|ENST00000442234.6|||||NP_996820.1||RefSeq|||||||||||||,T|upstream_gene_variant|MODIFIER|ABHD13|84945|Transcript|XM_011521128.3|protein_coding||||||||||1|3380|1||SNV|EntrezGene|HGNC:20293|||||||XP_011519430.1||RefSeq|||||||||||||,T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000273230|promoter||||||||||1||||SNV||||||||||||||||||||||||,T|TF_binding_site_variant|MODIFIER|||MotifFeature|ENSM00000634950|||||||||||1||1||SNV||||||||||||||||ENSPFM0617|3|Y|-0.094|ZBTB7B&ZBTB7A&ZBTB7C||||\n\n\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:23)\n\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:23)\n\tat is.hail.utils.package$.fatal(package.scala:96)\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2106)\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7$adapted(LoadVCF.scala:2076)\n\tat scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:515)\n\tat scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor_region3_18(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m64split_Block(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$10(BackendUtils.scala:90)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:166)\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$9(BackendUtils.scala:89)\n\tat is.hail.backend.spark.SparkBackend$$anon$5.compute(SparkBackend.scala:444)\n\tat org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)\n\tat org.apache.spark.rdd.RDD.iterator(RDD.scala:331)\n\tat org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)\n\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)\n\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\n\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)\n\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)\n\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\n\nis.hail.utils.HailException: Invalid locus 'chr13:108215366' found. Contig 'chr13' is not in the reference genome 'GRCh37'.\n\tat is.hail.utils.ErrorHandling.fatal(ErrorHandling.scala:19)\n\tat is.hail.utils.ErrorHandling.fatal$(ErrorHandling.scala:19)\n\tat is.hail.utils.package$.fatal(package.scala:96)\n\tat is.hail.variant.ReferenceGenome.checkLocus(ReferenceGenome.scala:237)\n\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3(LoadVCF.scala:521)\n\tat is.hail.io.vcf.VCFLine.$anonfun$parseAddVariant$3$adapted(LoadVCF.scala:521)\n\tat scala.Option.foreach(Option.scala:407)\n\tat is.hail.io.vcf.VCFLine.parseAddVariant(LoadVCF.scala:521)\n\tat is.hail.io.vcf.LoadVCF$.parseLine(LoadVCF.scala:1483)\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7(LoadVCF.scala:2092)\n\tat is.hail.io.vcf.MatrixVCFReader.$anonfun$executeGeneric$7$adapted(LoadVCF.scala:2076)\n\tat scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:515)\n\tat scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor_region3_18(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m134split_StreamFor(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.__m64split_Block(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n\tat __C52collect_distributed_array_table_coerce_sortedness.apply(Unknown Source)\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$10(BackendUtils.scala:90)\n\tat is.hail.utils.package$.using(package.scala:673)\n\tat is.hail.annotations.RegionPool.scopedRegion(RegionPool.scala:166)\n\tat is.hail.backend.BackendUtils.$anonfun$collectDArray$9(BackendUtils.scala:89)\n\tat is.hail.backend.spark.SparkBackend$$anon$5.compute(SparkBackend.scala:444)\n\tat org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)\n\tat org.apache.spark.rdd.RDD.iterator(RDD.scala:331)\n\tat org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)\n\tat org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)\n\tat org.apache.spark.scheduler.Task.run(Task.scala:141)\n\tat org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)\n\tat org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)\n\tat org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)\n\tat org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\n\n\n\n\n\nHail version: 0.2.133-4c60fddb171a\nError summary: HailException: Invalid locus 'chr13:108215366' found. Contig 'chr13' is not in the reference genome 'GRCh37'."
     ]
    }
   ],
   "source": [
    "print(f\"Initial dataset dimensions:\")\n",
    "print(f\"- Number of variants: {mt.count_rows():,}\")\n",
    "print(f\"- Number of samples: {mt.count_cols():,}\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "genome-env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
